# Day-2 : Data Manipulation and Visualization in R

---

## Learning Objectives

By the end of this day, students will be able to:

- Understand tidy data principles
- Use `dplyr` for data manipulation
- Create publication-quality plots using `ggplot2`
- Work efficiently with real-world datasets
- Perform basic exploratory data analysis (EDA)

---

## Introduction to the Tidyverse

The **tidyverse** is a collection of R packages designed for:

- Data manipulation
- Data visualization
- Data analysis workflows

Core tidyverse packages include:

- `ggplot2` – data visualization  
- `dplyr` – data manipulation  
- `tidyr` – data reshaping  
- `readr`  – data import  
- `tibble` – modern data frames  
- `stringr` – string manipulation  
- `forcats` – categorical data handling  
- `lubridate` - makes it easier to work with dates and times
- `readxl` - data import * 

note: readxl is not loaded automatically with library(tidyverse)

---

## Installing and Loading Packages

Install packages (only once):

```{r, eval=FALSE}
# install.packages("tidyverse")
```

Load packages (before every session):

```{r}

library(tidyverse)

```

## Tidy Data Principles

A dataset is tidy if:

- Each variable is a column

- Each observation is a row

- Each value is a cell

These principles make data easier to manipulate, visualize and analyze. 

### Working with a Real Dataset

We will use the built-in iris dataset.

```{r}
head(iris)
str(iris)

```

## The Pipe Operator (`%>%`)

When working with the tidyverse, we often perform multiple operations in sequence.
The **pipe operator (`%>%`)** allows us to write these steps in a clear and readable way.

The pipe takes the output of one step and passes it as the input to the next step. You can read `%>%` as “and then”, which helps in understanding the flow of the code.


Without the pipe

```{r}
iris[1:6, c("Sepal.Length", "Sepal.Width")]

```

with pipe 
```{r}

iris %>%
  select(Sepal.Length, Sepal.Width) %>%
  head()

```

The pipe allows code to be read from top to bottom, similar to a workflow.


## Data Manipulation with dplyr

Note: Most tidyverse functions assume that data is already in tidy format.


Selecting Columns, no need to write "" when writing name of columns

```{r}
iris %>% select(Sepal.Length, Sepal.Width) %>% head(10)

```

Exclude columns:

```{r}
iris %>% select(-Species) %>% head()
```

Filtering Rows

```{r}
iris %>% filter(Species == "setosa") %>% head()

```

Multiple conditions:

```{r}
iris %>% filter(Sepal.Length > 5, Species == "virginica") %>% head()

```

Creating New Variables (mutate)

```{r}
iris %>%
mutate(
Sepal.Area = Sepal.Length * Sepal.Width
) %>% head()

```

Sorting Data (arrange)

ascending: 

```{r}

iris %>% arrange(Sepal.Length)%>% head()

```

descending

```{r}
iris %>% arrange(desc(Sepal.Length))%>% head()
```

Summarizing Data (summarise)

```{r}
iris %>%
  summarise(
  mean_sepal_length = mean(Sepal.Length),
  sd_sepal_length = sd(Sepal.Length)
  )

```

Grouped Operations (group_by)

This allow us to perform summaries separately for each group in the data.


```{r}
iris %>%
  group_by(Species) %>%
  summarise(
    mean_sepal_length = mean(Sepal.Length),
    mean_petal_length = mean(Petal.Length)
  )

```

rename() — Rename Columns

Used to make column names shorter or more meaningful.

```{r}
iris %>%
  rename(
    SL = Sepal.Length,
    SW = Sepal.Width
  ) %>%
  head()

```

distinct() — Keep Unique Rows

Used to remove duplicate rows or extract unique values.

```{r}

iris %>%
  distinct(Species)

iris %>%
  distinct(Species, Sepal.Length)

```

slice() — Select Rows by Position

Used when you want rows based on row numbers, not conditions.

```{r}

iris %>%
  slice(1:5)

iris %>%
  slice((n() - 4):n())

```

pull() — Extract a Column as a Vector

Converts a data frame column into a plain vector.

```{r}
species_vec <- iris %>%
  pull(Species)

species_vec

```

sample_n() — Randomly Sample Rows

Randomly select a fixed number of rows.

```{r}
iris %>%
  sample_n(10)

```

sample_frac() — Sample a Fraction of Rows

Randomly select a fraction of the dataset.

```{r}
iris %>%
  sample_frac(0.1)  # 10% of the data

```


## Joins

Joining datasets is common in biology (samples + metadata

```{r}
# Example datasets
df1 <- tibble(ID = 1:3, Value = c(10,20,30))
df2 <- tibble(ID = 2:4, Group = c("A","B","C"))

# Inner join
inner_join(df1, df2, by = "ID")

# Left join
left_join(df1, df2, by = "ID")
# right join
right_join(df1, df2, by = "ID")

# full join

df <- full_join(df1, df2, by = "ID") 
df %>% drop_na() # removing all nas dplyr function

# drop by condition
df %>% 
  filter(!is.na(Value))

df %>% 
  filter(!is.na(Group))

df %>% 
  filter(!is.na(Group) | !is.na(Value))

df %>% 
  filter(!is.na(Group) & !is.na(Value))

```

### Reshaping Data (tidyr)

```{r}

library(tidyr)
library(tibble)

gene_expression_wide <- tibble(
  Gene = c("BRCA1", "TP53", "EGFR"),
  Sample_A = c(12.5, 8.3, 15.2),
  Sample_B = c(13.1, 9.0, 14.8),
  Sample_C = c(11.9, 8.7, 16.0)
)

gene_expression_wide

# Each row = a gene
# Each column = a sample
# This is easy for humans to read but not tidy.

# Wideto Long (Tidy Format)

# Convert sample columns into rows using pivot_longer().

gene_expression_long <- gene_expression_wide %>%
  pivot_longer(
    cols = starts_with("Sample"),
    names_to = "Sample",
    values_to = "Expression"
  )

gene_expression_long
# easy inetegration with ggplot format

ggplot(gene_expression_long,
       aes(x = Sample, y = Expression, fill = Gene)) +
  geom_col(position = "dodge") +
  labs(
    title = "Gene Expression Across Samples",
    y = "Expression Level"
  )

# Long to Wide (When Needed)

# Sometimes, you need to go back to wide format (e.g., for export or specific tools).

gene_expression_wide_again <- gene_expression_long %>%
  pivot_wider(
    names_from = Sample,
    values_from = Expression
  )

gene_expression_wide_again


# time dependent data

time_course <- tibble(
  Gene = c("MYC", "MYC", "MYC",
           "TP53", "TP53", "TP53",
           "EGFR", "EGFR", "EGFR"),
  Time = c("0h", "6h", "12h",
           "0h", "6h", "12h",
           "0h", "6h", "12h"),
  Expression = c(
    5.2, 8.1, 12.4,   # MYC
    7.5, 6.8, 5.9,    # TP53
    10.1, 12.3, 14.8  # EGFR
  )
)

time_course

time_course %>%
  pivot_wider(
    names_from = Time,
    values_from = Expression
  )

ggplot(time_course,
       aes(x = Time, y = Expression, color = Gene, group = Gene)) +
  geom_line() +
  geom_point() +
  labs(
    title = "Gene Expression Time Course",
    y = "Expression Level"
  )


```

When Column Names Are Badly Named

Suppose the dataset came from an external source with inconsistent naming:

```{r}
gene_expression_bad <- tibble(
  `Gene ID` = c("BRCA1", "TP53", "EGFR"),
  `Sample A (day 0)` = c(12.5, 8.3, 15.2),
  `Sample-B day6` = c(13.1, 9.0, 14.8),
  `sample_c_day_12` = c(11.9, 8.7, 16.0)
)

gene_expression_bad

# Problems:
# Spaces and special characters
# Inconsistent capitalization
# Hard to use in code

# Cleaning Column Names (Recommended Workflow)
# Rename Manually Using rename()

gene_expression_clean <- gene_expression_bad %>%
  rename(
    Gene = `Gene ID`,
    Sample_A = `Sample A (day 0)`,
    Sample_B = `Sample-B day6`,
    Sample_C = `sample_c_day_12`
  )

gene_expression_clean

# Use this when:
# Dataset is small
# You want full control over names

# Clean Names Automatically (Common in Practice)
library(janitor)

gene_expression_auto <- gene_expression_bad %>%
  janitor::clean_names()

gene_expression_auto





```

Converting Columns to Rows (Wide → Long)

```{r}

gene_expression_wide <- tibble(
  Gene = c("BRCA1", "TP53", "EGFR"),
  Sample_A = c(12.5, 8.3, 15.2),
  Sample_B = c(13.1, 9.0, 14.8),
  Sample_C = c(11.9, 8.7, 16.0)
)

gene_expression_wide

# Step 1: Convert to Matrix (Required for Transpose)
# Transposition works on matrices, not tibbles.

expr_matrix <- gene_expression_wide %>%
  column_to_rownames("Gene") %>%
  as.matrix()

expr_matrix

expr_matrix_t <- t(expr_matrix)
expr_matrix_t
expr_df <- as.data.frame(expr_matrix_t)
expr_df

# keep row names as column, often help ful
expr_df %>%
  rownames_to_column("Sample")

```


## Introduction to ggplot2

ggplot2 is based on the Grammar of Graphics, which means plots are built by combining independent components.

Core Components of a ggplot

- Data: the dataset you want to visualize
- Aesthetics (aes): how variables map to visual properties (x, y, color, size, etc.)
- Geometries (geom_*): the type of plot (points, lines, bars, boxes, …)


Basic structure:

```{r}
# ggplot(data, aes(x = variable1, y = variable2)) +
#   geom_*

```

### Basic Scatter Plot

Relationship Between Two Variables
Question answered:
Are two numeric variables related?

```{r}
ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width)) +
geom_point()

```

Adding Color and Labels

```{r}

ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width, color = Species)) +
geom_point() +
labs(
title = "Sepal Length vs Width",
x = "Sepal Length",
y = "Sepal Width"
)

# Use case (biology):
# Comparing morphological measurements across species groups.

```

### Scatter Plot with Trend Line

Question answered:
Is there an overall trend or correlation?

```{r}
ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width)) +
geom_point() +
geom_smooth(method = "lm", se = FALSE)

# what if we fit regression group wise

ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width, color = Species)) +
geom_point() +
geom_smooth(method = "lm", se = FALSE)

# combine group and overall trend

ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width)) +
geom_point() +
geom_smooth(method = "lm", se = FALSE, color = "black") +
geom_smooth(
aes(color = Species),
method = "lm",
se = FALSE
)

# In biological data, different groups (species, treatments, conditions) often show different relationships.
# Grouped regression lines help reveal these group-specific patterns.

```


### Boxplots 

Compare Distributions Across Groups

Question answered:

How does a numeric variable differ between groups?

```{r}

ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
geom_boxplot()

# Use case (biology):
# Comparing gene expression, cell size, or phenotype across conditions.
```

### Histograms 

Distribution of a Single Variable

Question answered:

How is a variable distributed?


```{r}

ggplot(iris, aes(x = Sepal.Length)) +
geom_histogram(bins = 20)

# histogram by group
ggplot(iris, aes(x = Sepal.Length, fill = Species)) +
geom_histogram(bins = 20, alpha = 0.6)


```

### Density Plots 

Smooth Distribution

Comparing distributions without binning effects.

```{r}
ggplot(iris, aes(x = Sepal.Length)) +
geom_density()

# by group 

ggplot(iris, aes(x = Sepal.Length, color = Species)) +
geom_density()

```

Violin Plots 

Distribution Shape + Spread (density plot or histogram + boxplot)

```{r}
ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
geom_violin(trim = FALSE)

```

Bar Plots 

Counts and Summaries

Count of Observations per Group

```{r}

ggplot(iris, aes(x = Species)) +
geom_bar()

ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
  stat_summary(fun = mean, geom = "bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2)

```

Line Plots 

Trends Over Time

Example: simulated time-course data

```{r}
time_course <- tibble(
Time = c(0, 6, 12),
Expression = c(5.2, 8.1, 12.4)
)

ggplot(time_course, aes(x = Time, y = Expression)) +
geom_line() +
geom_point()

# Gene expression over time or treatment duration.

```

Faceting (Multiple Plots by Group)

```{r}

ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width)) +
geom_point() +
facet_wrap(~ Species)

# Use case:
# Compare patterns across experimental conditions side-by-side.

```

### Venn Diagrams

Used for:

Overlap of genes between experiments

Shared pathways

Common differentially expressed genes

```{r}

# if (!require(devtools)) install.packages("devtools")
# devtools::install_github("yanlinlin82/ggvenn")

set.seed(20190708)
genes <- paste("gene",1:1000,sep="")
x <- list(
  A = sample(genes,300), 
  B = sample(genes,525), 
  C = sample(genes,440),
  D = sample(genes,350)
  )

library(ggvenn)

ggvenn(
  x, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

```

```{r}
library(pheatmap)

# Create a hypothetical gene expression matrix

expression_matrix <- matrix(
c(
8.2, 6.1, 7.4, 5.9,
3.2, 2.8, 3.5, 2.9,
10.5, 11.2, 9.8, 10.1,
4.6, 5.1, 4.8, 5.3,
7.9, 8.4, 8.1, 8.7
),
nrow = 5,
byrow = TRUE
)

# Add gene names (rows)

rownames(expression_matrix) <- c(
"BRCA1",
"TP53",
"EGFR",
"MYC",
"CDKN1A"
)

# Add sample names (columns)

colnames(expression_matrix) <- c(
"Sample_1",
"Sample_2",
"Sample_3",
"Sample_4"
)

expression_matrix

pheatmap(expression_matrix, cluster_rows = FALSE,
  cluster_cols = FALSE)

pheatmap(expression_matrix)

```



Combining dplyr and ggplot2

```{r}

iris %>%
  filter(Species != "setosa") %>%
  ggplot(aes(x = Petal.Length, y = Petal.Width, color = Species)) +
  geom_point()

```

## Exploratory Data Analysis (EDA)

Exploratory Data Analysis (EDA) is the process of examining and summarizing data to understand its main characteristics before applying formal statistical models.

EDA relies heavily on visualization and simple summaries, rather than statistical tests.

**Why EDA is important**

- EDA helps you to:

- Understand the structure and shape of the data

- Detect outliers and unusual observations

- Identify patterns and relationships

- Compare groups

- Assess data quality

- Generate hypotheses for further analysis

### Understanding the Structure of the Data

Before any analysis, always inspect the data.
```{r}

summary(iris)
```

This helps answer:

- What variables are present?

- Are they numeric or categorical?

- Are ranges reasonable?

### Checking for Missing Values

```{r}
colSums(is.na(iris))

# can you write this in tidyverse way? 
```

### Understanding Distributions

Histogram (distribution of a numeric variable)

```{r}

ggplot(iris, aes(x = Sepal.Length)) +
  geom_histogram(bins = 30) +
  labs(
  title = "Distribution of Sepal Length",
  x = "Sepal Length",
  y = "Count"
  )

```

Questions to ask:

- Is the distribution symmetric or skewed?
- Are there extreme values?

### Identifying Outliers

```{r}

ggplot(iris, aes(y = Sepal.Length)) +
  geom_boxplot() +
  labs(
  title = "Boxplot of Sepal Length",
  y = "Sepal Length"
  )

```

Boxplots are useful for spotting:

- Outliers
- Spread of the data
- Median values

### Comparing Groups 

Boxplot by group

```{r}
ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
geom_boxplot() +
labs(
title = "Sepal Length by Species",
x = "Species",
y = "Sepal Length"
)

```

This helps answer:

- Do groups differ in central tendency?
- Is variability similar across groups?

### Exploring Relationships Between Variables

Scatter plot

```{r}
ggplot(iris, aes(x = Petal.Length, y = Petal.Width)) +
  geom_point() +
  labs(
  title = "Petal Length vs Petal Width",
  x = "Petal Length",
  y = "Petal Width"
  )

```

Scatter plot with grouping

```{r}
ggplot(iris, aes(x = Petal.Length, y = Petal.Width, color = Species)) +
geom_point() +
labs(
title = "Petal Length vs Petal Width by Species"
)

```

Questions to ask:

- Are variables related?
- Do relationships differ by group?

### Simple Group Summaries

```{r}

iris %>%
  group_by(Species) %>%
  summarise(
  mean_sepal_length = mean(Sepal.Length),
  sd_sepal_length = sd(Sepal.Length)
  )


```

These summaries describe patterns, not statistical significance.

### EDA vs Statistical Analysis

EDA answers:

- What does the data look like?

Statistical analysis answers:

Are the observed differences statistically significant?

Formal hypothesis testing will be covered in week-3 in Statistics sessions. Will give introduction in next session from prospect of analysis.

**Good EDA helps you:**

- Understand your data
- Avoid incorrect assumptions
- Ask better scientific questions

## Hands-on Exercises

### Exercise 1

Using iris dataset:

- Filter only versicolor

- Select Sepal.Length and Petal.Length

### Exercise 2

Create a new column:

Petal.ratio = Petal.Length to Petal.Width

### Exercise 3

- Create a plot:

X-axis: Petal.Length

Y-axis: Petal.Width

Color by Species

### Exercise 4 (Challenge)

Calculate mean Sepal.Length for each species and visualize it using a bar plot.
