<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Day-2 : Data Manipulation and Visualization in R | R Programming for Statistical &amp; Genomic Data Analysis</title>
  <meta name="description" content="2 Day-2 : Data Manipulation and Visualization in R | R Programming for Statistical &amp; Genomic Data Analysis" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Day-2 : Data Manipulation and Visualization in R | R Programming for Statistical &amp; Genomic Data Analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Day-2 : Data Manipulation and Visualization in R | R Programming for Statistical &amp; Genomic Data Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="day-1-fundamentals-of-r-programming.html"/>
<link rel="next" href="day-3-statistical-analysis-using-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Day-0 : Course Overview</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#course-objectives"><i class="fa fa-check"></i><b>0.1</b> Course Objectives</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#what-this-course-is-not-about"><i class="fa fa-check"></i><b>0.2</b> What This Course Is <strong>Not</strong> About</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#target-audience"><i class="fa fa-check"></i><b>0.3</b> Target Audience</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#software-requirements"><i class="fa fa-check"></i><b>0.4</b> Software Requirements</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#course-structure"><i class="fa fa-check"></i><b>0.5</b> Course Structure</a></li>
<li class="chapter" data-level="0.6" data-path="index.html"><a href="index.html#how-to-use-this-book"><i class="fa fa-check"></i><b>0.6</b> How to Use This Book</a></li>
<li class="chapter" data-level="0.7" data-path="index.html"><a href="index.html#suggested-reading"><i class="fa fa-check"></i><b>0.7</b> Suggested Reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html"><i class="fa fa-check"></i>Day-0 : Installation &amp; Environment Setup (Linux)</a>
<ul>
<li class="chapter" data-level="0.8" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#learning-objectives"><i class="fa fa-check"></i><b>0.8</b> Learning Objectives</a></li>
<li class="chapter" data-level="0.9" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-r-ubuntudebian"><i class="fa fa-check"></i><b>0.9</b> Installing R (Ubuntu/Debian)</a></li>
<li class="chapter" data-level="0.10" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#you-can-also-follow-below-link-for-installations"><i class="fa fa-check"></i><b>0.10</b> You can also follow below link for installations</a></li>
<li class="chapter" data-level="0.11" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-rstudio-desktop-linux"><i class="fa fa-check"></i><b>0.11</b> Installing RStudio Desktop (Linux)</a>
<ul>
<li class="chapter" data-level="0.11.1" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-2-install-rstudio"><i class="fa fa-check"></i><b>0.11.1</b> Step 2: Install RStudio</a></li>
<li class="chapter" data-level="0.11.2" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-3-launch-rstudio"><i class="fa fa-check"></i><b>0.11.2</b> Step 3: Launch RStudio</a></li>
<li class="chapter" data-level="0.11.3" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-rstudio-desktop-windows"><i class="fa fa-check"></i><b>0.11.3</b> Installing RStudio Desktop (Windows)</a></li>
<li class="chapter" data-level="0.11.4" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-1-download-rstudio"><i class="fa fa-check"></i><b>0.11.4</b> Step 1: Download RStudio</a></li>
<li class="chapter" data-level="0.11.5" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-2-install-rstudio-1"><i class="fa fa-check"></i><b>0.11.5</b> Step 2: Install RStudio</a></li>
<li class="chapter" data-level="0.11.6" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-3-launch-rstudio-1"><i class="fa fa-check"></i><b>0.11.6</b> Step 3: Launch RStudio</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html"><i class="fa fa-check"></i><b>1</b> Day-1 : Fundamentals of R Programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#learning-objectives-1"><i class="fa fa-check"></i><b>1.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#what-is-r"><i class="fa fa-check"></i><b>1.2</b> What Is R?</a></li>
<li class="chapter" data-level="1.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#rstudio-interface-quick-tour"><i class="fa fa-check"></i><b>1.3</b> RStudio Interface (Quick Tour)</a></li>
<li class="chapter" data-level="1.4" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#creating-objects-in-r"><i class="fa fa-check"></i><b>1.4</b> Creating Objects in R</a></li>
<li class="chapter" data-level="1.5" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#attributes"><i class="fa fa-check"></i><b>1.5</b> Attributes</a></li>
<li class="chapter" data-level="1.6" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#using-r-as-a-calculator"><i class="fa fa-check"></i><b>1.6</b> Using R as a Calculator</a></li>
<li class="chapter" data-level="1.7" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#comments"><i class="fa fa-check"></i><b>1.7</b> Comments</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#documentation-comments-using"><i class="fa fa-check"></i><b>1.7.1</b> Documentation comments using #’</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#basic-data-types"><i class="fa fa-check"></i><b>1.8</b> Basic Data Types</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#summary-basic-r-data-types"><i class="fa fa-check"></i><b>1.8.1</b> Summary: Basic R Data Types</a></li>
</ul></li>
<li class="chapter" data-level="1.9" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#data-structures"><i class="fa fa-check"></i><b>1.9</b> Data Structures</a>
<ul>
<li class="chapter" data-level="1.9.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#summary-core-r-data-structures"><i class="fa fa-check"></i><b>1.9.1</b> Summary: Core R Data Structures</a></li>
<li class="chapter" data-level="1.9.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#vector"><i class="fa fa-check"></i><b>1.9.2</b> Vector</a></li>
<li class="chapter" data-level="1.9.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#lists"><i class="fa fa-check"></i><b>1.9.3</b> Lists</a></li>
<li class="chapter" data-level="1.9.4" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#matrices"><i class="fa fa-check"></i><b>1.9.4</b> Matrices</a></li>
<li class="chapter" data-level="1.9.5" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#factors"><i class="fa fa-check"></i><b>1.9.5</b> Factors</a></li>
<li class="chapter" data-level="1.9.6" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#missing-values"><i class="fa fa-check"></i><b>1.9.6</b> Missing values</a></li>
<li class="chapter" data-level="1.9.7" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#data-frames"><i class="fa fa-check"></i><b>1.9.7</b> Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="1.10" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#vectorization-thinking-in-vectors"><i class="fa fa-check"></i><b>1.10</b> Vectorization: Thinking in Vectors</a></li>
<li class="chapter" data-level="1.11" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#working-with-text-strings"><i class="fa fa-check"></i><b>1.11</b> Working with Text (Strings)</a></li>
<li class="chapter" data-level="1.12" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#control-structures"><i class="fa fa-check"></i><b>1.12</b> Control Structures</a>
<ul>
<li class="chapter" data-level="1.12.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#loop-control-break-and-next"><i class="fa fa-check"></i><b>1.12.1</b> Loop Control: break and next</a></li>
</ul></li>
<li class="chapter" data-level="1.13" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#functions"><i class="fa fa-check"></i><b>1.13</b> functions</a></li>
<li class="chapter" data-level="1.14" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#variable-scope-in-r-basics"><i class="fa fa-check"></i><b>1.14</b> Variable Scope in R (Basics)</a></li>
<li class="chapter" data-level="1.15" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#dates-and-time-in-r"><i class="fa fa-check"></i><b>1.15</b> Dates and Time in R</a></li>
<li class="chapter" data-level="1.16" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#loop-functions-in-r-apply-family"><i class="fa fa-check"></i><b>1.16</b> Loop Functions in R (Apply Family)</a></li>
<li class="chapter" data-level="1.17" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#getting-help-in-r"><i class="fa fa-check"></i><b>1.17</b> Getting Help in R</a></li>
<li class="chapter" data-level="1.18" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#hands-on-exercises"><i class="fa fa-check"></i><b>1.18</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="1.18.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-1"><i class="fa fa-check"></i><b>1.18.1</b> Exercise 1</a></li>
<li class="chapter" data-level="1.18.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-2"><i class="fa fa-check"></i><b>1.18.2</b> Exercise 2</a></li>
<li class="chapter" data-level="1.18.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-3"><i class="fa fa-check"></i><b>1.18.3</b> Exercise 3</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html"><i class="fa fa-check"></i><b>2</b> Day-2 : Data Manipulation and Visualization in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#learning-objectives-2"><i class="fa fa-check"></i><b>2.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-the-tidyverse"><i class="fa fa-check"></i><b>2.2</b> Introduction to the Tidyverse</a></li>
<li class="chapter" data-level="2.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#installing-and-loading-packages"><i class="fa fa-check"></i><b>2.3</b> Installing and Loading Packages</a></li>
<li class="chapter" data-level="2.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#tidy-data-principles"><i class="fa fa-check"></i><b>2.4</b> Tidy Data Principles</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#working-with-a-real-dataset"><i class="fa fa-check"></i><b>2.4.1</b> Working with a Real Dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#the-pipe-operator"><i class="fa fa-check"></i><b>2.5</b> The Pipe Operator (<code>%&gt;%</code>)</a></li>
<li class="chapter" data-level="2.6" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#data-manipulation-with-dplyr"><i class="fa fa-check"></i><b>2.6</b> Data Manipulation with dplyr</a></li>
<li class="chapter" data-level="2.7" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#joins"><i class="fa fa-check"></i><b>2.7</b> Joins</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#reshaping-data-tidyr"><i class="fa fa-check"></i><b>2.7.1</b> Reshaping Data (tidyr)</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-ggplot2"><i class="fa fa-check"></i><b>2.8</b> Introduction to ggplot2</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#basic-scatter-plot"><i class="fa fa-check"></i><b>2.8.1</b> Basic Scatter Plot</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exploratory-data-analysis-eda"><i class="fa fa-check"></i><b>2.9</b> Exploratory Data Analysis (EDA)</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-the-structure-of-the-data"><i class="fa fa-check"></i><b>2.9.1</b> Understanding the Structure of the Data</a></li>
<li class="chapter" data-level="2.9.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#checking-for-missing-values"><i class="fa fa-check"></i><b>2.9.2</b> Checking for Missing Values</a></li>
<li class="chapter" data-level="2.9.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-distributions"><i class="fa fa-check"></i><b>2.9.3</b> Understanding Distributions</a></li>
<li class="chapter" data-level="2.9.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#identifying-outliers"><i class="fa fa-check"></i><b>2.9.4</b> Identifying Outliers</a></li>
<li class="chapter" data-level="2.9.5" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#comparing-groups"><i class="fa fa-check"></i><b>2.9.5</b> Comparing Groups</a></li>
<li class="chapter" data-level="2.9.6" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exploring-relationships-between-variables"><i class="fa fa-check"></i><b>2.9.6</b> Exploring Relationships Between Variables</a></li>
<li class="chapter" data-level="2.9.7" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#simple-group-summaries"><i class="fa fa-check"></i><b>2.9.7</b> Simple Group Summaries</a></li>
<li class="chapter" data-level="2.9.8" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#eda-vs-statistical-analysis"><i class="fa fa-check"></i><b>2.9.8</b> EDA vs Statistical Analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#hands-on-exercises-1"><i class="fa fa-check"></i><b>2.10</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="2.10.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-1-1"><i class="fa fa-check"></i><b>2.10.1</b> Exercise 1</a></li>
<li class="chapter" data-level="2.10.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-2-1"><i class="fa fa-check"></i><b>2.10.2</b> Exercise 2</a></li>
<li class="chapter" data-level="2.10.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-3-1"><i class="fa fa-check"></i><b>2.10.3</b> Exercise 3</a></li>
<li class="chapter" data-level="2.10.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-4-challenge"><i class="fa fa-check"></i><b>2.10.4</b> Exercise 4 (Challenge)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html"><i class="fa fa-check"></i><b>3</b> Day-3 : Statistical Analysis Using R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#learning-objectives-3"><i class="fa fa-check"></i><b>3.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="3.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#types-of-data"><i class="fa fa-check"></i><b>3.2</b> Types of Data</a></li>
<li class="chapter" data-level="3.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#descriptive-statistics"><i class="fa fa-check"></i><b>3.3</b> Descriptive Statistics</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#numeric-summaries"><i class="fa fa-check"></i><b>3.3.1</b> Numeric Summaries</a></li>
<li class="chapter" data-level="3.3.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#grouped-summaries"><i class="fa fa-check"></i><b>3.3.2</b> Grouped summaries:</a></li>
<li class="chapter" data-level="3.3.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#visualizing-distributions"><i class="fa fa-check"></i><b>3.3.3</b> Visualizing Distributions</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#introduction-to-probability-conceptual"><i class="fa fa-check"></i><b>3.4</b> Introduction to Probability (Conceptual)</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#sampling-variation"><i class="fa fa-check"></i><b>3.4.1</b> Sampling Variation</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#introduction-to-hypothesis-testing"><i class="fa fa-check"></i><b>3.5</b> Introduction to Hypothesis Testing</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#key-concepts"><i class="fa fa-check"></i><b>3.5.1</b> Key concepts:</a></li>
<li class="chapter" data-level="3.5.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#t-test"><i class="fa fa-check"></i><b>3.5.2</b> t-test</a></li>
<li class="chapter" data-level="3.5.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>3.5.3</b> Two-Sample t-test</a></li>
<li class="chapter" data-level="3.5.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#non-parametric-test-wilcoxon-test"><i class="fa fa-check"></i><b>3.5.4</b> Non-Parametric Test: Wilcoxon Test</a></li>
<li class="chapter" data-level="3.5.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#analysis-of-variance-anova"><i class="fa fa-check"></i><b>3.5.5</b> Analysis of Variance (ANOVA)</a></li>
<li class="chapter" data-level="3.5.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#correlation-analysis"><i class="fa fa-check"></i><b>3.5.6</b> Correlation Analysis</a></li>
<li class="chapter" data-level="3.5.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#linear-regression"><i class="fa fa-check"></i><b>3.5.7</b> Linear Regression</a></li>
<li class="chapter" data-level="3.5.8" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#visualizing-the-regression"><i class="fa fa-check"></i><b>3.5.8</b> Visualizing the Regression</a></li>
<li class="chapter" data-level="3.5.9" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#interpreting-statistical-results"><i class="fa fa-check"></i><b>3.5.9</b> Interpreting Statistical Results</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#machine-learning-statistical-learning-a-gentle-introduction"><i class="fa fa-check"></i><b>3.6</b> Machine Learning (Statistical Learning): A Gentle Introduction</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#statistics-vs-machine-learning-conceptual-view"><i class="fa fa-check"></i><b>3.6.1</b> Statistics vs Machine Learning (Conceptual View)</a></li>
<li class="chapter" data-level="3.6.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#a-simple-example-linear-regression-as-machine-learning"><i class="fa fa-check"></i><b>3.6.2</b> A Simple Example: Linear Regression as Machine Learning</a></li>
<li class="chapter" data-level="3.6.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#evaluating-predictions"><i class="fa fa-check"></i><b>3.6.3</b> Evaluating Predictions</a></li>
<li class="chapter" data-level="3.6.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#types-of-machine-learning-tasks"><i class="fa fa-check"></i><b>3.6.4</b> Types of Machine Learning Tasks</a></li>
<li class="chapter" data-level="3.6.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#regression-tasks"><i class="fa fa-check"></i><b>3.6.5</b> Regression Tasks</a></li>
<li class="chapter" data-level="3.6.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#classification-tasks"><i class="fa fa-check"></i><b>3.6.6</b> Classification Tasks</a></li>
<li class="chapter" data-level="3.6.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#advanced-machine-learning-algorithms-overview"><i class="fa fa-check"></i><b>3.6.7</b> Advanced Machine Learning Algorithms (Overview)</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#hands-on-exercises-2"><i class="fa fa-check"></i><b>3.7</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-1-2"><i class="fa fa-check"></i><b>3.7.1</b> Exercise 1</a></li>
<li class="chapter" data-level="3.7.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-2-2"><i class="fa fa-check"></i><b>3.7.2</b> Exercise 2</a></li>
<li class="chapter" data-level="3.7.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-3-2"><i class="fa fa-check"></i><b>3.7.3</b> Exercise 3</a></li>
<li class="chapter" data-level="3.7.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-4-challenge-1"><i class="fa fa-check"></i><b>3.7.4</b> Exercise 4 (Challenge)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html"><i class="fa fa-check"></i><b>4</b> Day-4 : Genomic Data Exploration Using R - 1</a>
<ul>
<li class="chapter" data-level="4.1" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#learning-objectives-4"><i class="fa fa-check"></i><b>4.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="4.2" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#what-is-genomic-data"><i class="fa fa-check"></i><b>4.2</b> What Is Genomic Data?</a></li>
<li class="chapter" data-level="4.3" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#what-is-bioconductor"><i class="fa fa-check"></i><b>4.3</b> What Is Bioconductor?</a></li>
<li class="chapter" data-level="4.4" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#installing-bioconductor"><i class="fa fa-check"></i><b>4.4</b> Installing Bioconductor</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#log-transformation-for-exploration"><i class="fa fa-check"></i><b>4.4.1</b> Log Transformation for Exploration</a></li>
<li class="chapter" data-level="4.4.2" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#differential-expression-analysis-conceptual"><i class="fa fa-check"></i><b>4.4.2</b> Differential Expression Analysis (Conceptual)</a></li>
<li class="chapter" data-level="4.4.3" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#volcano-plot"><i class="fa fa-check"></i><b>4.4.3</b> Volcano Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="day-5-genomic-data-exploration-using-r---2.html"><a href="day-5-genomic-data-exploration-using-r---2.html"><i class="fa fa-check"></i><b>5</b> Day-5 : Genomic Data Exploration Using R - 2</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Programming for Statistical &amp; Genomic Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="day-2-data-manipulation-and-visualization-in-r" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Day-2 : Data Manipulation and Visualization in R<a href="day-2-data-manipulation-and-visualization-in-r.html#day-2-data-manipulation-and-visualization-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<hr />
<div id="learning-objectives-2" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Learning Objectives<a href="day-2-data-manipulation-and-visualization-in-r.html#learning-objectives-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>By the end of this day, students will be able to:</p>
<ul>
<li>Understand tidy data principles</li>
<li>Use <code>dplyr</code> for data manipulation</li>
<li>Create publication-quality plots using <code>ggplot2</code></li>
<li>Work efficiently with real-world datasets</li>
<li>Perform basic exploratory data analysis (EDA)</li>
</ul>
<hr />
</div>
<div id="introduction-to-the-tidyverse" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Introduction to the Tidyverse<a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-the-tidyverse" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <strong>tidyverse</strong> is a collection of R packages designed for:</p>
<ul>
<li>Data manipulation</li>
<li>Data visualization</li>
<li>Data analysis workflows</li>
</ul>
<p>Core tidyverse packages include:</p>
<ul>
<li><code>ggplot2</code> – data visualization<br />
</li>
<li><code>dplyr</code> – data manipulation<br />
</li>
<li><code>tidyr</code> – data reshaping<br />
</li>
<li><code>readr</code> – data import<br />
</li>
<li><code>tibble</code> – modern data frames<br />
</li>
<li><code>stringr</code> – string manipulation<br />
</li>
<li><code>forcats</code> – categorical data handling<br />
</li>
<li><code>lubridate</code> - makes it easier to work with dates and times</li>
<li><code>readxl</code> - data import *</li>
</ul>
<p>note: readxl is not loaded automatically with library(tidyverse)</p>
<hr />
</div>
<div id="installing-and-loading-packages" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Installing and Loading Packages<a href="day-2-data-manipulation-and-visualization-in-r.html#installing-and-loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Install packages (only once):</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb366-1" tabindex="-1"></a><span class="co"># install.packages(&quot;tidyverse&quot;)</span></span></code></pre></div>
<p>Load packages (before every session):</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb367-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.6
## ✔ forcats   1.0.1     ✔ stringr   1.6.0
## ✔ ggplot2   4.0.1     ✔ tibble    3.3.1
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.2
## ✔ purrr     1.2.1     
## ── Conflicts ──────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="tidy-data-principles" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Tidy Data Principles<a href="day-2-data-manipulation-and-visualization-in-r.html#tidy-data-principles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A dataset is tidy if:</p>
<ul>
<li><p>Each variable is a column</p></li>
<li><p>Each observation is a row</p></li>
<li><p>Each value is a cell</p></li>
</ul>
<p>These principles make data easier to manipulate, visualize and analyze.</p>
<div id="working-with-a-real-dataset" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Working with a Real Dataset<a href="day-2-data-manipulation-and-visualization-in-r.html#working-with-a-real-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will use the built-in iris dataset.</p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb369-1" tabindex="-1"></a><span class="fu">head</span>(iris)</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
## 5          5.0         3.6          1.4         0.2  setosa
## 6          5.4         3.9          1.7         0.4  setosa</code></pre>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb371-1" tabindex="-1"></a><span class="fu">str</span>(iris)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    150 obs. of  5 variables:
##  $ Sepal.Length: num  5.1 4.9 4.7 4.6 5 5.4 4.6 5 4.4 4.9 ...
##  $ Sepal.Width : num  3.5 3 3.2 3.1 3.6 3.9 3.4 3.4 2.9 3.1 ...
##  $ Petal.Length: num  1.4 1.4 1.3 1.5 1.4 1.7 1.4 1.5 1.4 1.5 ...
##  $ Petal.Width : num  0.2 0.2 0.2 0.2 0.2 0.4 0.3 0.2 0.2 0.1 ...
##  $ Species     : Factor w/ 3 levels &quot;setosa&quot;,&quot;versicolor&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</code></pre>
</div>
</div>
<div id="the-pipe-operator" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> The Pipe Operator (<code>%&gt;%</code>)<a href="day-2-data-manipulation-and-visualization-in-r.html#the-pipe-operator" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When working with the tidyverse, we often perform multiple operations in sequence.
The <strong>pipe operator (<code>%&gt;%</code>)</strong> allows us to write these steps in a clear and readable way.</p>
<p>The pipe takes the output of one step and passes it as the input to the next step. You can read <code>%&gt;%</code> as “and then”, which helps in understanding the flow of the code.</p>
<p>Without the pipe</p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb373-1" tabindex="-1"></a>iris[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="fu">c</span>(<span class="st">&quot;Sepal.Length&quot;</span>, <span class="st">&quot;Sepal.Width&quot;</span>)]</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width
## 1          5.1         3.5
## 2          4.9         3.0
## 3          4.7         3.2
## 4          4.6         3.1
## 5          5.0         3.6
## 6          5.4         3.9</code></pre>
<p>with pipe</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb375-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb375-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb375-2" tabindex="-1"></a>  <span class="fu">select</span>(Sepal.Length, Sepal.Width) <span class="sc">%&gt;%</span></span>
<span id="cb375-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb375-3" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width
## 1          5.1         3.5
## 2          4.9         3.0
## 3          4.7         3.2
## 4          4.6         3.1
## 5          5.0         3.6
## 6          5.4         3.9</code></pre>
<p>The pipe allows code to be read from top to bottom, similar to a workflow.</p>
</div>
<div id="data-manipulation-with-dplyr" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Data Manipulation with dplyr<a href="day-2-data-manipulation-and-visualization-in-r.html#data-manipulation-with-dplyr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Note: Most tidyverse functions assume that data is already in tidy format.</p>
<p>Selecting Columns, no need to write “” when writing name of columns</p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb377-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">select</span>(Sepal.Length, Sepal.Width) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##    Sepal.Length Sepal.Width
## 1           5.1         3.5
## 2           4.9         3.0
## 3           4.7         3.2
## 4           4.6         3.1
## 5           5.0         3.6
## 6           5.4         3.9
## 7           4.6         3.4
## 8           5.0         3.4
## 9           4.4         2.9
## 10          4.9         3.1</code></pre>
<p>Exclude columns:</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb379-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Species) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width
## 1          5.1         3.5          1.4         0.2
## 2          4.9         3.0          1.4         0.2
## 3          4.7         3.2          1.3         0.2
## 4          4.6         3.1          1.5         0.2
## 5          5.0         3.6          1.4         0.2
## 6          5.4         3.9          1.7         0.4</code></pre>
<p>Filtering Rows</p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb381-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Species <span class="sc">==</span> <span class="st">&quot;setosa&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
## 5          5.0         3.6          1.4         0.2  setosa
## 6          5.4         3.9          1.7         0.4  setosa</code></pre>
<p>Multiple conditions:</p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb383-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sepal.Length <span class="sc">&gt;</span> <span class="dv">5</span>, Species <span class="sc">==</span> <span class="st">&quot;virginica&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width   Species
## 1          6.3         3.3          6.0         2.5 virginica
## 2          5.8         2.7          5.1         1.9 virginica
## 3          7.1         3.0          5.9         2.1 virginica
## 4          6.3         2.9          5.6         1.8 virginica
## 5          6.5         3.0          5.8         2.2 virginica
## 6          7.6         3.0          6.6         2.1 virginica</code></pre>
<p>Creating New Variables (mutate)</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb385-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb385-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb385-2" tabindex="-1"></a><span class="fu">mutate</span>(</span>
<span id="cb385-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb385-3" tabindex="-1"></a><span class="at">Sepal.Area =</span> Sepal.Length <span class="sc">*</span> Sepal.Width</span>
<span id="cb385-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb385-4" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species Sepal.Area
## 1          5.1         3.5          1.4         0.2  setosa      17.85
## 2          4.9         3.0          1.4         0.2  setosa      14.70
## 3          4.7         3.2          1.3         0.2  setosa      15.04
## 4          4.6         3.1          1.5         0.2  setosa      14.26
## 5          5.0         3.6          1.4         0.2  setosa      18.00
## 6          5.4         3.9          1.7         0.4  setosa      21.06</code></pre>
<p>Sorting Data (arrange)</p>
<p>ascending:</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb387-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(Sepal.Length)<span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          4.3         3.0          1.1         0.1  setosa
## 2          4.4         2.9          1.4         0.2  setosa
## 3          4.4         3.0          1.3         0.2  setosa
## 4          4.4         3.2          1.3         0.2  setosa
## 5          4.5         2.3          1.3         0.3  setosa
## 6          4.6         3.1          1.5         0.2  setosa</code></pre>
<p>descending</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb389-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Sepal.Length))<span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width   Species
## 1          7.9         3.8          6.4         2.0 virginica
## 2          7.7         3.8          6.7         2.2 virginica
## 3          7.7         2.6          6.9         2.3 virginica
## 4          7.7         2.8          6.7         2.0 virginica
## 5          7.7         3.0          6.1         2.3 virginica
## 6          7.6         3.0          6.6         2.1 virginica</code></pre>
<p>Summarizing Data (summarise)</p>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb391-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb391-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb391-2" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb391-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb391-3" tabindex="-1"></a>  <span class="at">mean_sepal_length =</span> <span class="fu">mean</span>(Sepal.Length),</span>
<span id="cb391-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb391-4" tabindex="-1"></a>  <span class="at">sd_sepal_length =</span> <span class="fu">sd</span>(Sepal.Length)</span>
<span id="cb391-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb391-5" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>##   mean_sepal_length sd_sepal_length
## 1          5.843333       0.8280661</code></pre>
<p>Grouped Operations (group_by)</p>
<p>This allow us to perform summaries separately for each group in the data.</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb393-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-2" tabindex="-1"></a>  <span class="fu">group_by</span>(Species) <span class="sc">%&gt;%</span></span>
<span id="cb393-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb393-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-4" tabindex="-1"></a>    <span class="at">mean_sepal_length =</span> <span class="fu">mean</span>(Sepal.Length),</span>
<span id="cb393-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-5" tabindex="-1"></a>    <span class="at">mean_petal_length =</span> <span class="fu">mean</span>(Petal.Length)</span>
<span id="cb393-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb393-6" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##   Species    mean_sepal_length mean_petal_length
##   &lt;fct&gt;                  &lt;dbl&gt;             &lt;dbl&gt;
## 1 setosa                  5.01              1.46
## 2 versicolor              5.94              4.26
## 3 virginica               6.59              5.55</code></pre>
<p>rename() — Rename Columns</p>
<p>Used to make column names shorter or more meaningful.</p>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb395-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-2" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb395-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-3" tabindex="-1"></a>    <span class="at">SL =</span> Sepal.Length,</span>
<span id="cb395-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-4" tabindex="-1"></a>    <span class="at">SW =</span> Sepal.Width</span>
<span id="cb395-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-5" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb395-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb395-6" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##    SL  SW Petal.Length Petal.Width Species
## 1 5.1 3.5          1.4         0.2  setosa
## 2 4.9 3.0          1.4         0.2  setosa
## 3 4.7 3.2          1.3         0.2  setosa
## 4 4.6 3.1          1.5         0.2  setosa
## 5 5.0 3.6          1.4         0.2  setosa
## 6 5.4 3.9          1.7         0.4  setosa</code></pre>
<p>distinct() — Keep Unique Rows</p>
<p>Used to remove duplicate rows or extract unique values.</p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb397-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb397-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb397-2" tabindex="-1"></a>  <span class="fu">distinct</span>(Species)</span></code></pre></div>
<pre><code>##      Species
## 1     setosa
## 2 versicolor
## 3  virginica</code></pre>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb399-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb399-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb399-2" tabindex="-1"></a>  <span class="fu">distinct</span>(Species, Sepal.Length)</span></code></pre></div>
<pre><code>##       Species Sepal.Length
## 1      setosa          5.1
## 2      setosa          4.9
## 3      setosa          4.7
## 4      setosa          4.6
## 5      setosa          5.0
## 6      setosa          5.4
## 7      setosa          4.4
## 8      setosa          4.8
## 9      setosa          4.3
## 10     setosa          5.8
## 11     setosa          5.7
## 12     setosa          5.2
## 13     setosa          5.5
## 14     setosa          4.5
## 15     setosa          5.3
## 16 versicolor          7.0
## 17 versicolor          6.4
## 18 versicolor          6.9
## 19 versicolor          5.5
## 20 versicolor          6.5
## 21 versicolor          5.7
## 22 versicolor          6.3
## 23 versicolor          4.9
## 24 versicolor          6.6
## 25 versicolor          5.2
## 26 versicolor          5.0
## 27 versicolor          5.9
## 28 versicolor          6.0
## 29 versicolor          6.1
## 30 versicolor          5.6
## 31 versicolor          6.7
## 32 versicolor          5.8
## 33 versicolor          6.2
## 34 versicolor          6.8
## 35 versicolor          5.4
## 36 versicolor          5.1
## 37  virginica          6.3
## 38  virginica          5.8
## 39  virginica          7.1
## 40  virginica          6.5
## 41  virginica          7.6
## 42  virginica          4.9
## 43  virginica          7.3
## 44  virginica          6.7
## 45  virginica          7.2
## 46  virginica          6.4
## 47  virginica          6.8
## 48  virginica          5.7
## 49  virginica          7.7
## 50  virginica          6.0
## 51  virginica          6.9
## 52  virginica          5.6
## 53  virginica          6.2
## 54  virginica          6.1
## 55  virginica          7.4
## 56  virginica          7.9
## 57  virginica          5.9</code></pre>
<p>slice() — Select Rows by Position</p>
<p>Used when you want rows based on row numbers, not conditions.</p>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb401-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb401-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb401-2" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
## 5          5.0         3.6          1.4         0.2  setosa</code></pre>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb403-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb403-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb403-2" tabindex="-1"></a>  <span class="fu">slice</span>((<span class="fu">n</span>() <span class="sc">-</span> <span class="dv">4</span>)<span class="sc">:</span><span class="fu">n</span>())</span></code></pre></div>
<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width   Species
## 1          6.7         3.0          5.2         2.3 virginica
## 2          6.3         2.5          5.0         1.9 virginica
## 3          6.5         3.0          5.2         2.0 virginica
## 4          6.2         3.4          5.4         2.3 virginica
## 5          5.9         3.0          5.1         1.8 virginica</code></pre>
<p>pull() — Extract a Column as a Vector</p>
<p>Converts a data frame column into a plain vector.</p>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb405-1" tabindex="-1"></a>species_vec <span class="ot">&lt;-</span> iris <span class="sc">%&gt;%</span></span>
<span id="cb405-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb405-2" tabindex="-1"></a>  <span class="fu">pull</span>(Species)</span>
<span id="cb405-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb405-3" tabindex="-1"></a></span>
<span id="cb405-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb405-4" tabindex="-1"></a>species_vec</span></code></pre></div>
<pre><code>##   [1] setosa     setosa     setosa     setosa     setosa     setosa    
##   [7] setosa     setosa     setosa     setosa     setosa     setosa    
##  [13] setosa     setosa     setosa     setosa     setosa     setosa    
##  [19] setosa     setosa     setosa     setosa     setosa     setosa    
##  [25] setosa     setosa     setosa     setosa     setosa     setosa    
##  [31] setosa     setosa     setosa     setosa     setosa     setosa    
##  [37] setosa     setosa     setosa     setosa     setosa     setosa    
##  [43] setosa     setosa     setosa     setosa     setosa     setosa    
##  [49] setosa     setosa     versicolor versicolor versicolor versicolor
##  [55] versicolor versicolor versicolor versicolor versicolor versicolor
##  [61] versicolor versicolor versicolor versicolor versicolor versicolor
##  [67] versicolor versicolor versicolor versicolor versicolor versicolor
##  [73] versicolor versicolor versicolor versicolor versicolor versicolor
##  [79] versicolor versicolor versicolor versicolor versicolor versicolor
##  [85] versicolor versicolor versicolor versicolor versicolor versicolor
##  [91] versicolor versicolor versicolor versicolor versicolor versicolor
##  [97] versicolor versicolor versicolor versicolor virginica  virginica 
## [103] virginica  virginica  virginica  virginica  virginica  virginica 
## [109] virginica  virginica  virginica  virginica  virginica  virginica 
## [115] virginica  virginica  virginica  virginica  virginica  virginica 
## [121] virginica  virginica  virginica  virginica  virginica  virginica 
## [127] virginica  virginica  virginica  virginica  virginica  virginica 
## [133] virginica  virginica  virginica  virginica  virginica  virginica 
## [139] virginica  virginica  virginica  virginica  virginica  virginica 
## [145] virginica  virginica  virginica  virginica  virginica  virginica 
## Levels: setosa versicolor virginica</code></pre>
<p>sample_n() — Randomly Sample Rows</p>
<p>Randomly select a fixed number of rows.</p>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb407-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb407-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb407-2" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##    Sepal.Length Sepal.Width Petal.Length Petal.Width    Species
## 1           6.3         2.3          4.4         1.3 versicolor
## 2           6.9         3.1          5.4         2.1  virginica
## 3           4.6         3.2          1.4         0.2     setosa
## 4           5.7         2.9          4.2         1.3 versicolor
## 5           5.6         3.0          4.5         1.5 versicolor
## 6           6.3         3.3          4.7         1.6 versicolor
## 7           6.5         2.8          4.6         1.5 versicolor
## 8           7.0         3.2          4.7         1.4 versicolor
## 9           5.7         3.0          4.2         1.2 versicolor
## 10          5.8         2.7          5.1         1.9  virginica</code></pre>
<p>sample_frac() — Sample a Fraction of Rows</p>
<p>Randomly select a fraction of the dataset.</p>
<div class="sourceCode" id="cb409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb409-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb409-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb409-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb409-2" tabindex="-1"></a>  <span class="fu">sample_frac</span>(<span class="fl">0.1</span>)  <span class="co"># 10% of the data</span></span></code></pre></div>
<pre><code>##    Sepal.Length Sepal.Width Petal.Length Petal.Width    Species
## 1           4.9         3.0          1.4         0.2     setosa
## 2           4.6         3.1          1.5         0.2     setosa
## 3           7.7         2.8          6.7         2.0  virginica
## 4           6.0         3.4          4.5         1.6 versicolor
## 5           6.7         3.3          5.7         2.5  virginica
## 6           4.6         3.2          1.4         0.2     setosa
## 7           7.2         3.6          6.1         2.5  virginica
## 8           7.7         3.8          6.7         2.2  virginica
## 9           6.4         2.7          5.3         1.9  virginica
## 10          4.8         3.4          1.9         0.2     setosa
## 11          6.8         3.2          5.9         2.3  virginica
## 12          7.7         3.0          6.1         2.3  virginica
## 13          4.4         3.2          1.3         0.2     setosa
## 14          6.9         3.2          5.7         2.3  virginica
## 15          6.3         2.7          4.9         1.8  virginica</code></pre>
</div>
<div id="joins" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Joins<a href="day-2-data-manipulation-and-visualization-in-r.html#joins" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Joining datasets is common in biology (samples + metadata</p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-1" tabindex="-1"></a><span class="co"># Example datasets</span></span>
<span id="cb411-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-2" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">ID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">Value =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">30</span>))</span>
<span id="cb411-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-3" tabindex="-1"></a>df2 <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">ID =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">Group =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;C&quot;</span>))</span>
<span id="cb411-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-4" tabindex="-1"></a></span>
<span id="cb411-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-5" tabindex="-1"></a><span class="co"># Inner join</span></span>
<span id="cb411-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb411-6" tabindex="-1"></a><span class="fu">inner_join</span>(df1, df2, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 2 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     2    20 A    
## 2     3    30 B</code></pre>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb413-1" tabindex="-1"></a><span class="co"># Left join</span></span>
<span id="cb413-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb413-2" tabindex="-1"></a><span class="fu">left_join</span>(df1, df2, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     1    10 &lt;NA&gt; 
## 2     2    20 A    
## 3     3    30 B</code></pre>
<div class="sourceCode" id="cb415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb415-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb415-1" tabindex="-1"></a><span class="co"># right join</span></span>
<span id="cb415-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb415-2" tabindex="-1"></a><span class="fu">right_join</span>(df1, df2, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     2    20 A    
## 2     3    30 B    
## 3     4    NA C</code></pre>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb417-1" tabindex="-1"></a><span class="co"># full join</span></span>
<span id="cb417-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb417-2" tabindex="-1"></a></span>
<span id="cb417-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb417-3" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">full_join</span>(df1, df2, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>) </span>
<span id="cb417-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb417-4" tabindex="-1"></a>df <span class="sc">%&gt;%</span> <span class="fu">drop_na</span>() <span class="co"># removing all nas dplyr function</span></span></code></pre></div>
<pre><code>## # A tibble: 2 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     2    20 A    
## 2     3    30 B</code></pre>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb419-1" tabindex="-1"></a><span class="co"># drop by condition</span></span>
<span id="cb419-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb419-2" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb419-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb419-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Value))</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     1    10 &lt;NA&gt; 
## 2     2    20 A    
## 3     3    30 B</code></pre>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb421-1" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb421-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb421-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Group))</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     2    20 A    
## 2     3    30 B    
## 3     4    NA C</code></pre>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb423-1" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb423-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb423-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Group) <span class="sc">|</span> <span class="sc">!</span><span class="fu">is.na</span>(Value))</span></code></pre></div>
<pre><code>## # A tibble: 4 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     1    10 &lt;NA&gt; 
## 2     2    20 A    
## 3     3    30 B    
## 4     4    NA C</code></pre>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb425-1" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb425-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb425-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Group) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(Value))</span></code></pre></div>
<pre><code>## # A tibble: 2 × 3
##      ID Value Group
##   &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;
## 1     2    20 A    
## 2     3    30 B</code></pre>
<div id="reshaping-data-tidyr" class="section level3 hasAnchor" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Reshaping Data (tidyr)<a href="day-2-data-manipulation-and-visualization-in-r.html#reshaping-data-tidyr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-1" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb427-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-2" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb427-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-3" tabindex="-1"></a></span>
<span id="cb427-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-4" tabindex="-1"></a>gene_expression_wide <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb427-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-5" tabindex="-1"></a>  <span class="at">Gene =</span> <span class="fu">c</span>(<span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;EGFR&quot;</span>),</span>
<span id="cb427-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-6" tabindex="-1"></a>  <span class="at">Sample_A =</span> <span class="fu">c</span>(<span class="fl">12.5</span>, <span class="fl">8.3</span>, <span class="fl">15.2</span>),</span>
<span id="cb427-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-7" tabindex="-1"></a>  <span class="at">Sample_B =</span> <span class="fu">c</span>(<span class="fl">13.1</span>, <span class="fl">9.0</span>, <span class="fl">14.8</span>),</span>
<span id="cb427-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-8" tabindex="-1"></a>  <span class="at">Sample_C =</span> <span class="fu">c</span>(<span class="fl">11.9</span>, <span class="fl">8.7</span>, <span class="fl">16.0</span>)</span>
<span id="cb427-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-9" tabindex="-1"></a>)</span>
<span id="cb427-10"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-10" tabindex="-1"></a></span>
<span id="cb427-11"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb427-11" tabindex="-1"></a>gene_expression_wide</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   Gene  Sample_A Sample_B Sample_C
##   &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 BRCA1     12.5     13.1     11.9
## 2 TP53       8.3      9        8.7
## 3 EGFR      15.2     14.8     16</code></pre>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-1" tabindex="-1"></a><span class="co"># Each row = a gene</span></span>
<span id="cb429-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-2" tabindex="-1"></a><span class="co"># Each column = a sample</span></span>
<span id="cb429-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-3" tabindex="-1"></a><span class="co"># This is easy for humans to read but not tidy.</span></span>
<span id="cb429-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-4" tabindex="-1"></a></span>
<span id="cb429-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-5" tabindex="-1"></a><span class="co"># Wideto Long (Tidy Format)</span></span>
<span id="cb429-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-6" tabindex="-1"></a></span>
<span id="cb429-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-7" tabindex="-1"></a><span class="co"># Convert sample columns into rows using pivot_longer().</span></span>
<span id="cb429-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-8" tabindex="-1"></a></span>
<span id="cb429-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-9" tabindex="-1"></a>gene_expression_long <span class="ot">&lt;-</span> gene_expression_wide <span class="sc">%&gt;%</span></span>
<span id="cb429-10"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-10" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb429-11"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-11" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">&quot;Sample&quot;</span>),</span>
<span id="cb429-12"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-12" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;Sample&quot;</span>,</span>
<span id="cb429-13"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-13" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;Expression&quot;</span></span>
<span id="cb429-14"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-14" tabindex="-1"></a>  )</span>
<span id="cb429-15"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-15" tabindex="-1"></a></span>
<span id="cb429-16"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb429-16" tabindex="-1"></a>gene_expression_long</span></code></pre></div>
<pre><code>## # A tibble: 9 × 3
##   Gene  Sample   Expression
##   &lt;chr&gt; &lt;chr&gt;         &lt;dbl&gt;
## 1 BRCA1 Sample_A       12.5
## 2 BRCA1 Sample_B       13.1
## 3 BRCA1 Sample_C       11.9
## 4 TP53  Sample_A        8.3
## 5 TP53  Sample_B        9  
## 6 TP53  Sample_C        8.7
## 7 EGFR  Sample_A       15.2
## 8 EGFR  Sample_B       14.8
## 9 EGFR  Sample_C       16</code></pre>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-1" tabindex="-1"></a><span class="co"># easy inetegration with ggplot format</span></span>
<span id="cb431-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-2" tabindex="-1"></a></span>
<span id="cb431-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-3" tabindex="-1"></a><span class="fu">ggplot</span>(gene_expression_long,</span>
<span id="cb431-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-4" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Expression, <span class="at">fill =</span> Gene)) <span class="sc">+</span></span>
<span id="cb431-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-5" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="sc">+</span></span>
<span id="cb431-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-6" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb431-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-7" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Gene Expression Across Samples&quot;</span>,</span>
<span id="cb431-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-8" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Expression Level&quot;</span></span>
<span id="cb431-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb431-9" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-82-1.png" alt="" width="672" /></p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-1" tabindex="-1"></a><span class="co"># Long to Wide (When Needed)</span></span>
<span id="cb432-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-2" tabindex="-1"></a></span>
<span id="cb432-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-3" tabindex="-1"></a><span class="co"># Sometimes, you need to go back to wide format (e.g., for export or specific tools).</span></span>
<span id="cb432-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-4" tabindex="-1"></a></span>
<span id="cb432-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-5" tabindex="-1"></a>gene_expression_wide_again <span class="ot">&lt;-</span> gene_expression_long <span class="sc">%&gt;%</span></span>
<span id="cb432-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-6" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb432-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-7" tabindex="-1"></a>    <span class="at">names_from =</span> Sample,</span>
<span id="cb432-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-8" tabindex="-1"></a>    <span class="at">values_from =</span> Expression</span>
<span id="cb432-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-9" tabindex="-1"></a>  )</span>
<span id="cb432-10"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-10" tabindex="-1"></a></span>
<span id="cb432-11"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb432-11" tabindex="-1"></a>gene_expression_wide_again</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   Gene  Sample_A Sample_B Sample_C
##   &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 BRCA1     12.5     13.1     11.9
## 2 TP53       8.3      9        8.7
## 3 EGFR      15.2     14.8     16</code></pre>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-1" tabindex="-1"></a><span class="co"># time dependent data</span></span>
<span id="cb434-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-2" tabindex="-1"></a></span>
<span id="cb434-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-3" tabindex="-1"></a>time_course <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb434-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-4" tabindex="-1"></a>  <span class="at">Gene =</span> <span class="fu">c</span>(<span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb434-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-5" tabindex="-1"></a>           <span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb434-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-6" tabindex="-1"></a>           <span class="st">&quot;EGFR&quot;</span>, <span class="st">&quot;EGFR&quot;</span>, <span class="st">&quot;EGFR&quot;</span>),</span>
<span id="cb434-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-7" tabindex="-1"></a>  <span class="at">Time =</span> <span class="fu">c</span>(<span class="st">&quot;0h&quot;</span>, <span class="st">&quot;6h&quot;</span>, <span class="st">&quot;12h&quot;</span>,</span>
<span id="cb434-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-8" tabindex="-1"></a>           <span class="st">&quot;0h&quot;</span>, <span class="st">&quot;6h&quot;</span>, <span class="st">&quot;12h&quot;</span>,</span>
<span id="cb434-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-9" tabindex="-1"></a>           <span class="st">&quot;0h&quot;</span>, <span class="st">&quot;6h&quot;</span>, <span class="st">&quot;12h&quot;</span>),</span>
<span id="cb434-10"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-10" tabindex="-1"></a>  <span class="at">Expression =</span> <span class="fu">c</span>(</span>
<span id="cb434-11"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-11" tabindex="-1"></a>    <span class="fl">5.2</span>, <span class="fl">8.1</span>, <span class="fl">12.4</span>,   <span class="co"># MYC</span></span>
<span id="cb434-12"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-12" tabindex="-1"></a>    <span class="fl">7.5</span>, <span class="fl">6.8</span>, <span class="fl">5.9</span>,    <span class="co"># TP53</span></span>
<span id="cb434-13"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-13" tabindex="-1"></a>    <span class="fl">10.1</span>, <span class="fl">12.3</span>, <span class="fl">14.8</span>  <span class="co"># EGFR</span></span>
<span id="cb434-14"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-14" tabindex="-1"></a>  )</span>
<span id="cb434-15"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-15" tabindex="-1"></a>)</span>
<span id="cb434-16"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-16" tabindex="-1"></a></span>
<span id="cb434-17"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb434-17" tabindex="-1"></a>time_course</span></code></pre></div>
<pre><code>## # A tibble: 9 × 3
##   Gene  Time  Expression
##   &lt;chr&gt; &lt;chr&gt;      &lt;dbl&gt;
## 1 MYC   0h           5.2
## 2 MYC   6h           8.1
## 3 MYC   12h         12.4
## 4 TP53  0h           7.5
## 5 TP53  6h           6.8
## 6 TP53  12h          5.9
## 7 EGFR  0h          10.1
## 8 EGFR  6h          12.3
## 9 EGFR  12h         14.8</code></pre>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb436-1" tabindex="-1"></a>time_course <span class="sc">%&gt;%</span></span>
<span id="cb436-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb436-2" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb436-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb436-3" tabindex="-1"></a>    <span class="at">names_from =</span> Time,</span>
<span id="cb436-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb436-4" tabindex="-1"></a>    <span class="at">values_from =</span> Expression</span>
<span id="cb436-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb436-5" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   Gene   `0h`  `6h` `12h`
##   &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 MYC     5.2   8.1  12.4
## 2 TP53    7.5   6.8   5.9
## 3 EGFR   10.1  12.3  14.8</code></pre>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-1" tabindex="-1"></a><span class="fu">ggplot</span>(time_course,</span>
<span id="cb438-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-2" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> Time, <span class="at">y =</span> Expression, <span class="at">color =</span> Gene, <span class="at">group =</span> Gene)) <span class="sc">+</span></span>
<span id="cb438-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-3" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb438-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-4" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb438-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-5" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb438-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-6" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Gene Expression Time Course&quot;</span>,</span>
<span id="cb438-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-7" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Expression Level&quot;</span></span>
<span id="cb438-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb438-8" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-82-2.png" alt="" width="672" /></p>
<p>When Column Names Are Badly Named</p>
<p>Suppose the dataset came from an external source with inconsistent naming:</p>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-1" tabindex="-1"></a>gene_expression_bad <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb439-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-2" tabindex="-1"></a>  <span class="st">`</span><span class="at">Gene ID</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;EGFR&quot;</span>),</span>
<span id="cb439-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sample A (day 0)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">12.5</span>, <span class="fl">8.3</span>, <span class="fl">15.2</span>),</span>
<span id="cb439-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Sample-B day6</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">13.1</span>, <span class="fl">9.0</span>, <span class="fl">14.8</span>),</span>
<span id="cb439-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">sample_c_day_12</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">11.9</span>, <span class="fl">8.7</span>, <span class="fl">16.0</span>)</span>
<span id="cb439-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-6" tabindex="-1"></a>)</span>
<span id="cb439-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-7" tabindex="-1"></a></span>
<span id="cb439-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb439-8" tabindex="-1"></a>gene_expression_bad</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   `Gene ID` `Sample A (day 0)` `Sample-B day6` sample_c_day_12
##   &lt;chr&gt;                  &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;
## 1 BRCA1                   12.5            13.1            11.9
## 2 TP53                     8.3             9               8.7
## 3 EGFR                    15.2            14.8            16</code></pre>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-1" tabindex="-1"></a><span class="co"># Problems:</span></span>
<span id="cb441-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-2" tabindex="-1"></a><span class="co"># Spaces and special characters</span></span>
<span id="cb441-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-3" tabindex="-1"></a><span class="co"># Inconsistent capitalization</span></span>
<span id="cb441-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-4" tabindex="-1"></a><span class="co"># Hard to use in code</span></span>
<span id="cb441-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-5" tabindex="-1"></a></span>
<span id="cb441-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-6" tabindex="-1"></a><span class="co"># Cleaning Column Names (Recommended Workflow)</span></span>
<span id="cb441-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-7" tabindex="-1"></a><span class="co"># Rename Manually Using rename()</span></span>
<span id="cb441-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-8" tabindex="-1"></a></span>
<span id="cb441-9"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-9" tabindex="-1"></a>gene_expression_clean <span class="ot">&lt;-</span> gene_expression_bad <span class="sc">%&gt;%</span></span>
<span id="cb441-10"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-10" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb441-11"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-11" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="st">`</span><span class="at">Gene ID</span><span class="st">`</span>,</span>
<span id="cb441-12"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-12" tabindex="-1"></a>    <span class="at">Sample_A =</span> <span class="st">`</span><span class="at">Sample A (day 0)</span><span class="st">`</span>,</span>
<span id="cb441-13"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-13" tabindex="-1"></a>    <span class="at">Sample_B =</span> <span class="st">`</span><span class="at">Sample-B day6</span><span class="st">`</span>,</span>
<span id="cb441-14"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-14" tabindex="-1"></a>    <span class="at">Sample_C =</span> <span class="st">`</span><span class="at">sample_c_day_12</span><span class="st">`</span></span>
<span id="cb441-15"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-15" tabindex="-1"></a>  )</span>
<span id="cb441-16"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-16" tabindex="-1"></a></span>
<span id="cb441-17"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb441-17" tabindex="-1"></a>gene_expression_clean</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   Gene  Sample_A Sample_B Sample_C
##   &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 BRCA1     12.5     13.1     11.9
## 2 TP53       8.3      9        8.7
## 3 EGFR      15.2     14.8     16</code></pre>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-1" tabindex="-1"></a><span class="co"># Use this when:</span></span>
<span id="cb443-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-2" tabindex="-1"></a><span class="co"># Dataset is small</span></span>
<span id="cb443-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-3" tabindex="-1"></a><span class="co"># You want full control over names</span></span>
<span id="cb443-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-4" tabindex="-1"></a></span>
<span id="cb443-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-5" tabindex="-1"></a><span class="co"># Clean Names Automatically (Common in Practice)</span></span>
<span id="cb443-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb443-6" tabindex="-1"></a><span class="fu">library</span>(janitor)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;janitor&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     chisq.test, fisher.test</code></pre>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb446-1" tabindex="-1"></a>gene_expression_auto <span class="ot">&lt;-</span> gene_expression_bad <span class="sc">%&gt;%</span></span>
<span id="cb446-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb446-2" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>()</span>
<span id="cb446-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb446-3" tabindex="-1"></a></span>
<span id="cb446-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb446-4" tabindex="-1"></a>gene_expression_auto</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   gene_id sample_a_day_0 sample_b_day6 sample_c_day_12
##   &lt;chr&gt;            &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;
## 1 BRCA1             12.5          13.1            11.9
## 2 TP53               8.3           9               8.7
## 3 EGFR              15.2          14.8            16</code></pre>
<p>Converting Columns to Rows (Wide → Long)</p>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-1" tabindex="-1"></a>gene_expression_wide <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb448-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-2" tabindex="-1"></a>  <span class="at">Gene =</span> <span class="fu">c</span>(<span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;EGFR&quot;</span>),</span>
<span id="cb448-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-3" tabindex="-1"></a>  <span class="at">Sample_A =</span> <span class="fu">c</span>(<span class="fl">12.5</span>, <span class="fl">8.3</span>, <span class="fl">15.2</span>),</span>
<span id="cb448-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-4" tabindex="-1"></a>  <span class="at">Sample_B =</span> <span class="fu">c</span>(<span class="fl">13.1</span>, <span class="fl">9.0</span>, <span class="fl">14.8</span>),</span>
<span id="cb448-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-5" tabindex="-1"></a>  <span class="at">Sample_C =</span> <span class="fu">c</span>(<span class="fl">11.9</span>, <span class="fl">8.7</span>, <span class="fl">16.0</span>)</span>
<span id="cb448-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-6" tabindex="-1"></a>)</span>
<span id="cb448-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-7" tabindex="-1"></a></span>
<span id="cb448-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb448-8" tabindex="-1"></a>gene_expression_wide</span></code></pre></div>
<pre><code>## # A tibble: 3 × 4
##   Gene  Sample_A Sample_B Sample_C
##   &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 BRCA1     12.5     13.1     11.9
## 2 TP53       8.3      9        8.7
## 3 EGFR      15.2     14.8     16</code></pre>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-1" tabindex="-1"></a><span class="co"># Step 1: Convert to Matrix (Required for Transpose)</span></span>
<span id="cb450-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-2" tabindex="-1"></a><span class="co"># Transposition works on matrices, not tibbles.</span></span>
<span id="cb450-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-3" tabindex="-1"></a></span>
<span id="cb450-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-4" tabindex="-1"></a>expr_matrix <span class="ot">&lt;-</span> gene_expression_wide <span class="sc">%&gt;%</span></span>
<span id="cb450-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-5" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;Gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb450-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-6" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb450-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-7" tabindex="-1"></a></span>
<span id="cb450-8"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb450-8" tabindex="-1"></a>expr_matrix</span></code></pre></div>
<pre><code>##       Sample_A Sample_B Sample_C
## BRCA1     12.5     13.1     11.9
## TP53       8.3      9.0      8.7
## EGFR      15.2     14.8     16.0</code></pre>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb452-1" tabindex="-1"></a>expr_matrix_t <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_matrix)</span>
<span id="cb452-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb452-2" tabindex="-1"></a>expr_matrix_t</span></code></pre></div>
<pre><code>##          BRCA1 TP53 EGFR
## Sample_A  12.5  8.3 15.2
## Sample_B  13.1  9.0 14.8
## Sample_C  11.9  8.7 16.0</code></pre>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb454-1" tabindex="-1"></a>expr_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(expr_matrix_t)</span>
<span id="cb454-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb454-2" tabindex="-1"></a>expr_df</span></code></pre></div>
<pre><code>##          BRCA1 TP53 EGFR
## Sample_A  12.5  8.3 15.2
## Sample_B  13.1  9.0 14.8
## Sample_C  11.9  8.7 16.0</code></pre>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb456-1" tabindex="-1"></a><span class="co"># keep row names as column, often help ful</span></span>
<span id="cb456-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb456-2" tabindex="-1"></a>expr_df <span class="sc">%&gt;%</span></span>
<span id="cb456-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb456-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>)</span></code></pre></div>
<pre><code>##     Sample BRCA1 TP53 EGFR
## 1 Sample_A  12.5  8.3 15.2
## 2 Sample_B  13.1  9.0 14.8
## 3 Sample_C  11.9  8.7 16.0</code></pre>
</div>
</div>
<div id="introduction-to-ggplot2" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Introduction to ggplot2<a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-ggplot2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The grammar of graphics:</p>
<ul>
<li>Data</li>
<li>Aesthetics (aes)</li>
<li>Geometries (geom_*)</li>
</ul>
<p>In ggplot2, plots are built by combining data, aesthetics, and geometric layers.</p>
<div id="basic-scatter-plot" class="section level3 hasAnchor" number="2.8.1">
<h3><span class="header-section-number">2.8.1</span> Basic Scatter Plot<a href="day-2-data-manipulation-and-visualization-in-r.html#basic-scatter-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb458-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length, <span class="at">y =</span> Sepal.Width)) <span class="sc">+</span></span>
<span id="cb458-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb458-2" tabindex="-1"></a><span class="fu">geom_point</span>()</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-85-1.png" alt="" width="672" /></p>
<p>Adding Color and Labels</p>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length, <span class="at">y =</span> Sepal.Width, <span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb459-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-2" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb459-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-3" tabindex="-1"></a><span class="fu">labs</span>(</span>
<span id="cb459-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-4" tabindex="-1"></a><span class="at">title =</span> <span class="st">&quot;Sepal Length vs Width&quot;</span>,</span>
<span id="cb459-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-5" tabindex="-1"></a><span class="at">x =</span> <span class="st">&quot;Sepal Length&quot;</span>,</span>
<span id="cb459-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-6" tabindex="-1"></a><span class="at">y =</span> <span class="st">&quot;Sepal Width&quot;</span></span>
<span id="cb459-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb459-7" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-86-1.png" alt="" width="672" /></p>
<p>Boxplots</p>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb460-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Species, <span class="at">y =</span> Sepal.Length, <span class="at">fill =</span> Species)) <span class="sc">+</span></span>
<span id="cb460-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb460-2" tabindex="-1"></a><span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-87-1.png" alt="" width="672" /></p>
<p>Histograms</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb461-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length)) <span class="sc">+</span></span>
<span id="cb461-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb461-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-88-1.png" alt="" width="672" /></p>
<p>Combining dplyr and ggplot2</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb462-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb462-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb462-2" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">&quot;setosa&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb462-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb462-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Petal.Length, <span class="at">y =</span> Petal.Width, <span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb462-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb462-4" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-89-1.png" alt="" width="672" /></p>
</div>
</div>
<div id="exploratory-data-analysis-eda" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Exploratory Data Analysis (EDA)<a href="day-2-data-manipulation-and-visualization-in-r.html#exploratory-data-analysis-eda" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Exploratory Data Analysis (EDA) is the process of examining and summarizing data to understand its main characteristics before applying formal statistical models.</p>
<p>EDA relies heavily on visualization and simple summaries, rather than statistical tests.</p>
<p><strong>Why EDA is important</strong></p>
<ul>
<li><p>EDA helps you to:</p></li>
<li><p>Understand the structure and shape of the data</p></li>
<li><p>Detect outliers and unusual observations</p></li>
<li><p>Identify patterns and relationships</p></li>
<li><p>Compare groups</p></li>
<li><p>Assess data quality</p></li>
<li><p>Generate hypotheses for further analysis</p></li>
</ul>
<div id="understanding-the-structure-of-the-data" class="section level3 hasAnchor" number="2.9.1">
<h3><span class="header-section-number">2.9.1</span> Understanding the Structure of the Data<a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-the-structure-of-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before any analysis, always inspect the data.</p>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb463-1" tabindex="-1"></a><span class="fu">summary</span>(iris)</span></code></pre></div>
<pre><code>##   Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   
##  Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  
##  1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  
##  Median :5.800   Median :3.000   Median :4.350   Median :1.300  
##  Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  
##  3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  
##  Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  
##        Species  
##  setosa    :50  
##  versicolor:50  
##  virginica :50  
##                 
##                 
## </code></pre>
<p>This helps answer:</p>
<ul>
<li><p>What variables are present?</p></li>
<li><p>Are they numeric or categorical?</p></li>
<li><p>Are ranges reasonable?</p></li>
</ul>
</div>
<div id="checking-for-missing-values" class="section level3 hasAnchor" number="2.9.2">
<h3><span class="header-section-number">2.9.2</span> Checking for Missing Values<a href="day-2-data-manipulation-and-visualization-in-r.html#checking-for-missing-values" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb465-1" tabindex="-1"></a><span class="fu">colSums</span>(<span class="fu">is.na</span>(iris))</span></code></pre></div>
<pre><code>## Sepal.Length  Sepal.Width Petal.Length  Petal.Width      Species 
##            0            0            0            0            0</code></pre>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb467-1" tabindex="-1"></a><span class="co"># can you write this in tidyverse way? </span></span></code></pre></div>
</div>
<div id="understanding-distributions" class="section level3 hasAnchor" number="2.9.3">
<h3><span class="header-section-number">2.9.3</span> Understanding Distributions<a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-distributions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Histogram (distribution of a numeric variable)</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length)) <span class="sc">+</span></span>
<span id="cb468-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb468-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-3" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb468-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-4" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">&quot;Distribution of Sepal Length&quot;</span>,</span>
<span id="cb468-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-5" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">&quot;Sepal Length&quot;</span>,</span>
<span id="cb468-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-6" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">&quot;Count&quot;</span></span>
<span id="cb468-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb468-7" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-92-1.png" alt="" width="672" /></p>
<p>Questions to ask:</p>
<ul>
<li>Is the distribution symmetric or skewed?</li>
<li>Are there extreme values?</li>
</ul>
</div>
<div id="identifying-outliers" class="section level3 hasAnchor" number="2.9.4">
<h3><span class="header-section-number">2.9.4</span> Identifying Outliers<a href="day-2-data-manipulation-and-visualization-in-r.html#identifying-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">y =</span> Sepal.Length)) <span class="sc">+</span></span>
<span id="cb469-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-2" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb469-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-3" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb469-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-4" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">&quot;Boxplot of Sepal Length&quot;</span>,</span>
<span id="cb469-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-5" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">&quot;Sepal Length&quot;</span></span>
<span id="cb469-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb469-6" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-93-1.png" alt="" width="672" /></p>
<p>Boxplots are useful for spotting:</p>
<ul>
<li>Outliers</li>
<li>Spread of the data</li>
<li>Median values</li>
</ul>
</div>
<div id="comparing-groups" class="section level3 hasAnchor" number="2.9.5">
<h3><span class="header-section-number">2.9.5</span> Comparing Groups<a href="day-2-data-manipulation-and-visualization-in-r.html#comparing-groups" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Boxplot by group</p>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Species, <span class="at">y =</span> Sepal.Length, <span class="at">fill =</span> Species)) <span class="sc">+</span></span>
<span id="cb470-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-2" tabindex="-1"></a><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb470-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-3" tabindex="-1"></a><span class="fu">labs</span>(</span>
<span id="cb470-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-4" tabindex="-1"></a><span class="at">title =</span> <span class="st">&quot;Sepal Length by Species&quot;</span>,</span>
<span id="cb470-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-5" tabindex="-1"></a><span class="at">x =</span> <span class="st">&quot;Species&quot;</span>,</span>
<span id="cb470-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-6" tabindex="-1"></a><span class="at">y =</span> <span class="st">&quot;Sepal Length&quot;</span></span>
<span id="cb470-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb470-7" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-94-1.png" alt="" width="672" /></p>
<p>This helps answer:</p>
<ul>
<li>Do groups differ in central tendency?</li>
<li>Is variability similar across groups?</li>
</ul>
</div>
<div id="exploring-relationships-between-variables" class="section level3 hasAnchor" number="2.9.6">
<h3><span class="header-section-number">2.9.6</span> Exploring Relationships Between Variables<a href="day-2-data-manipulation-and-visualization-in-r.html#exploring-relationships-between-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Scatter plot</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Petal.Length, <span class="at">y =</span> Petal.Width)) <span class="sc">+</span></span>
<span id="cb471-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb471-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-3" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb471-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-4" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">&quot;Petal Length vs Petal Width&quot;</span>,</span>
<span id="cb471-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-5" tabindex="-1"></a>  <span class="at">x =</span> <span class="st">&quot;Petal Length&quot;</span>,</span>
<span id="cb471-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-6" tabindex="-1"></a>  <span class="at">y =</span> <span class="st">&quot;Petal Width&quot;</span></span>
<span id="cb471-7"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb471-7" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-95-1.png" alt="" width="672" /></p>
<p>Scatter plot with grouping</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb472-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x =</span> Petal.Length, <span class="at">y =</span> Petal.Width, <span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb472-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb472-2" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb472-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb472-3" tabindex="-1"></a><span class="fu">labs</span>(</span>
<span id="cb472-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb472-4" tabindex="-1"></a><span class="at">title =</span> <span class="st">&quot;Petal Length vs Petal Width by Species&quot;</span></span>
<span id="cb472-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb472-5" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-96-1.png" alt="" width="672" /></p>
<p>Questions to ask:</p>
<ul>
<li>Are variables related?</li>
<li>Do relationships differ by group?</li>
</ul>
</div>
<div id="simple-group-summaries" class="section level3 hasAnchor" number="2.9.7">
<h3><span class="header-section-number">2.9.7</span> Simple Group Summaries<a href="day-2-data-manipulation-and-visualization-in-r.html#simple-group-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-1" tabindex="-1"></a>iris <span class="sc">%&gt;%</span></span>
<span id="cb473-2"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-2" tabindex="-1"></a>  <span class="fu">group_by</span>(Species) <span class="sc">%&gt;%</span></span>
<span id="cb473-3"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb473-4"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-4" tabindex="-1"></a>  <span class="at">mean_sepal_length =</span> <span class="fu">mean</span>(Sepal.Length),</span>
<span id="cb473-5"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-5" tabindex="-1"></a>  <span class="at">sd_sepal_length =</span> <span class="fu">sd</span>(Sepal.Length)</span>
<span id="cb473-6"><a href="day-2-data-manipulation-and-visualization-in-r.html#cb473-6" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##   Species    mean_sepal_length sd_sepal_length
##   &lt;fct&gt;                  &lt;dbl&gt;           &lt;dbl&gt;
## 1 setosa                  5.01           0.352
## 2 versicolor              5.94           0.516
## 3 virginica               6.59           0.636</code></pre>
<p>These summaries describe patterns, not statistical significance.</p>
</div>
<div id="eda-vs-statistical-analysis" class="section level3 hasAnchor" number="2.9.8">
<h3><span class="header-section-number">2.9.8</span> EDA vs Statistical Analysis<a href="day-2-data-manipulation-and-visualization-in-r.html#eda-vs-statistical-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>EDA answers:</p>
<ul>
<li>What does the data look like?</li>
</ul>
<p>Statistical analysis answers:</p>
<p>Are the observed differences statistically significant?</p>
<p>Formal hypothesis testing will be covered in week-3 in Statistics sessions. Will give introduction in next session from prospect of analysis.</p>
<p><strong>Good EDA helps you:</strong></p>
<ul>
<li>Understand your data</li>
<li>Avoid incorrect assumptions</li>
<li>Ask better scientific questions</li>
</ul>
</div>
</div>
<div id="hands-on-exercises-1" class="section level2 hasAnchor" number="2.10">
<h2><span class="header-section-number">2.10</span> Hands-on Exercises<a href="day-2-data-manipulation-and-visualization-in-r.html#hands-on-exercises-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="exercise-1-1" class="section level3 hasAnchor" number="2.10.1">
<h3><span class="header-section-number">2.10.1</span> Exercise 1<a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Using iris dataset:</p>
<ul>
<li><p>Filter only versicolor</p></li>
<li><p>Select Sepal.Length and Petal.Length</p></li>
</ul>
</div>
<div id="exercise-2-1" class="section level3 hasAnchor" number="2.10.2">
<h3><span class="header-section-number">2.10.2</span> Exercise 2<a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create a new column:</p>
<p>Petal.ratio = Petal.Length to Petal.Width</p>
</div>
<div id="exercise-3-1" class="section level3 hasAnchor" number="2.10.3">
<h3><span class="header-section-number">2.10.3</span> Exercise 3<a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-3-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Create a plot:</li>
</ul>
<p>X-axis: Petal.Length</p>
<p>Y-axis: Petal.Width</p>
<p>Color by Species</p>
</div>
<div id="exercise-4-challenge" class="section level3 hasAnchor" number="2.10.4">
<h3><span class="header-section-number">2.10.4</span> Exercise 4 (Challenge)<a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-4-challenge" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Calculate mean Sepal.Length for each species and visualize it using a bar plot.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="day-1-fundamentals-of-r-programming.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="day-3-statistical-analysis-using-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["R_Programming_Course.pdf", "R_Programming_Course.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  }
});
});
</script>

</body>

</html>
