<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Day-4 : Genomic Data Exploration Using R - 1 | R Programming for Statistical &amp; Genomic Data Analysis</title>
  <meta name="description" content="4 Day-4 : Genomic Data Exploration Using R - 1 | R Programming for Statistical &amp; Genomic Data Analysis" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Day-4 : Genomic Data Exploration Using R - 1 | R Programming for Statistical &amp; Genomic Data Analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Day-4 : Genomic Data Exploration Using R - 1 | R Programming for Statistical &amp; Genomic Data Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="day-3-statistical-analysis-using-r.html"/>
<link rel="next" href="day-5-genomic-data-exploration-using-r---2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Day-0 : Course Overview</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#course-objectives"><i class="fa fa-check"></i><b>0.1</b> Course Objectives</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#what-this-course-is-not-about"><i class="fa fa-check"></i><b>0.2</b> What This Course Is <strong>Not</strong> About</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#target-audience"><i class="fa fa-check"></i><b>0.3</b> Target Audience</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#software-requirements"><i class="fa fa-check"></i><b>0.4</b> Software Requirements</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#course-structure"><i class="fa fa-check"></i><b>0.5</b> Course Structure</a></li>
<li class="chapter" data-level="0.6" data-path="index.html"><a href="index.html#how-to-use-this-book"><i class="fa fa-check"></i><b>0.6</b> How to Use This Book</a></li>
<li class="chapter" data-level="0.7" data-path="index.html"><a href="index.html#suggested-reading"><i class="fa fa-check"></i><b>0.7</b> Suggested Reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html"><i class="fa fa-check"></i>Day-0 : Installation &amp; Environment Setup (Linux)</a>
<ul>
<li class="chapter" data-level="0.8" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#learning-objectives"><i class="fa fa-check"></i><b>0.8</b> Learning Objectives</a></li>
<li class="chapter" data-level="0.9" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-r-ubuntudebian"><i class="fa fa-check"></i><b>0.9</b> Installing R (Ubuntu/Debian)</a></li>
<li class="chapter" data-level="0.10" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#you-can-also-follow-below-link-for-installations"><i class="fa fa-check"></i><b>0.10</b> You can also follow below link for installations</a></li>
<li class="chapter" data-level="0.11" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-rstudio-desktop-linux"><i class="fa fa-check"></i><b>0.11</b> Installing RStudio Desktop (Linux)</a>
<ul>
<li class="chapter" data-level="0.11.1" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-2-install-rstudio"><i class="fa fa-check"></i><b>0.11.1</b> Step 2: Install RStudio</a></li>
<li class="chapter" data-level="0.11.2" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-3-launch-rstudio"><i class="fa fa-check"></i><b>0.11.2</b> Step 3: Launch RStudio</a></li>
<li class="chapter" data-level="0.11.3" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#installing-rstudio-desktop-windows"><i class="fa fa-check"></i><b>0.11.3</b> Installing RStudio Desktop (Windows)</a></li>
<li class="chapter" data-level="0.11.4" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-1-download-rstudio"><i class="fa fa-check"></i><b>0.11.4</b> Step 1: Download RStudio</a></li>
<li class="chapter" data-level="0.11.5" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-2-install-rstudio-1"><i class="fa fa-check"></i><b>0.11.5</b> Step 2: Install RStudio</a></li>
<li class="chapter" data-level="0.11.6" data-path="day-0-installation-environment-setup-linux.html"><a href="day-0-installation-environment-setup-linux.html#step-3-launch-rstudio-1"><i class="fa fa-check"></i><b>0.11.6</b> Step 3: Launch RStudio</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html"><i class="fa fa-check"></i><b>1</b> Day-1 : Fundamentals of R Programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#learning-objectives-1"><i class="fa fa-check"></i><b>1.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#what-is-r"><i class="fa fa-check"></i><b>1.2</b> What Is R?</a></li>
<li class="chapter" data-level="1.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#rstudio-interface-quick-tour"><i class="fa fa-check"></i><b>1.3</b> RStudio Interface (Quick Tour)</a></li>
<li class="chapter" data-level="1.4" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#creating-objects-in-r"><i class="fa fa-check"></i><b>1.4</b> Creating Objects in R</a></li>
<li class="chapter" data-level="1.5" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#attributes"><i class="fa fa-check"></i><b>1.5</b> Attributes</a></li>
<li class="chapter" data-level="1.6" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#using-r-as-a-calculator"><i class="fa fa-check"></i><b>1.6</b> Using R as a Calculator</a></li>
<li class="chapter" data-level="1.7" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#comments"><i class="fa fa-check"></i><b>1.7</b> Comments</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#documentation-comments-using"><i class="fa fa-check"></i><b>1.7.1</b> Documentation comments using #’</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#basic-data-types"><i class="fa fa-check"></i><b>1.8</b> Basic Data Types</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#summary-basic-r-data-types"><i class="fa fa-check"></i><b>1.8.1</b> Summary: Basic R Data Types</a></li>
</ul></li>
<li class="chapter" data-level="1.9" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#data-structures"><i class="fa fa-check"></i><b>1.9</b> Data Structures</a>
<ul>
<li class="chapter" data-level="1.9.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#summary-core-r-data-structures"><i class="fa fa-check"></i><b>1.9.1</b> Summary: Core R Data Structures</a></li>
<li class="chapter" data-level="1.9.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#vector"><i class="fa fa-check"></i><b>1.9.2</b> Vector</a></li>
<li class="chapter" data-level="1.9.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#lists"><i class="fa fa-check"></i><b>1.9.3</b> Lists</a></li>
<li class="chapter" data-level="1.9.4" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#matrices"><i class="fa fa-check"></i><b>1.9.4</b> Matrices</a></li>
<li class="chapter" data-level="1.9.5" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#factors"><i class="fa fa-check"></i><b>1.9.5</b> Factors</a></li>
<li class="chapter" data-level="1.9.6" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#missing-values"><i class="fa fa-check"></i><b>1.9.6</b> Missing values</a></li>
<li class="chapter" data-level="1.9.7" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#data-frames"><i class="fa fa-check"></i><b>1.9.7</b> Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="1.10" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#vectorization-thinking-in-vectors"><i class="fa fa-check"></i><b>1.10</b> Vectorization: Thinking in Vectors</a></li>
<li class="chapter" data-level="1.11" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#working-with-text-strings"><i class="fa fa-check"></i><b>1.11</b> Working with Text (Strings)</a></li>
<li class="chapter" data-level="1.12" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#control-structures"><i class="fa fa-check"></i><b>1.12</b> Control Structures</a>
<ul>
<li class="chapter" data-level="1.12.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#loop-control-break-and-next"><i class="fa fa-check"></i><b>1.12.1</b> Loop Control: break and next</a></li>
</ul></li>
<li class="chapter" data-level="1.13" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#functions"><i class="fa fa-check"></i><b>1.13</b> functions</a></li>
<li class="chapter" data-level="1.14" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#variable-scope-in-r-basics"><i class="fa fa-check"></i><b>1.14</b> Variable Scope in R (Basics)</a></li>
<li class="chapter" data-level="1.15" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#dates-and-time-in-r"><i class="fa fa-check"></i><b>1.15</b> Dates and Time in R</a></li>
<li class="chapter" data-level="1.16" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#loop-functions-in-r-apply-family"><i class="fa fa-check"></i><b>1.16</b> Loop Functions in R (Apply Family)</a></li>
<li class="chapter" data-level="1.17" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#getting-help-in-r"><i class="fa fa-check"></i><b>1.17</b> Getting Help in R</a></li>
<li class="chapter" data-level="1.18" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#hands-on-exercises"><i class="fa fa-check"></i><b>1.18</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="1.18.1" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-1"><i class="fa fa-check"></i><b>1.18.1</b> Exercise 1</a></li>
<li class="chapter" data-level="1.18.2" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-2"><i class="fa fa-check"></i><b>1.18.2</b> Exercise 2</a></li>
<li class="chapter" data-level="1.18.3" data-path="day-1-fundamentals-of-r-programming.html"><a href="day-1-fundamentals-of-r-programming.html#exercise-3"><i class="fa fa-check"></i><b>1.18.3</b> Exercise 3</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html"><i class="fa fa-check"></i><b>2</b> Day-2 : Data Manipulation and Visualization in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#learning-objectives-2"><i class="fa fa-check"></i><b>2.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-the-tidyverse"><i class="fa fa-check"></i><b>2.2</b> Introduction to the Tidyverse</a></li>
<li class="chapter" data-level="2.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#installing-and-loading-packages"><i class="fa fa-check"></i><b>2.3</b> Installing and Loading Packages</a></li>
<li class="chapter" data-level="2.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#tidy-data-principles"><i class="fa fa-check"></i><b>2.4</b> Tidy Data Principles</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#working-with-a-real-dataset"><i class="fa fa-check"></i><b>2.4.1</b> Working with a Real Dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#the-pipe-operator"><i class="fa fa-check"></i><b>2.5</b> The Pipe Operator (<code>%&gt;%</code>)</a></li>
<li class="chapter" data-level="2.6" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#data-manipulation-with-dplyr"><i class="fa fa-check"></i><b>2.6</b> Data Manipulation with dplyr</a></li>
<li class="chapter" data-level="2.7" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#joins"><i class="fa fa-check"></i><b>2.7</b> Joins</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#reshaping-data-tidyr"><i class="fa fa-check"></i><b>2.7.1</b> Reshaping Data (tidyr)</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#introduction-to-ggplot2"><i class="fa fa-check"></i><b>2.8</b> Introduction to ggplot2</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#basic-scatter-plot"><i class="fa fa-check"></i><b>2.8.1</b> Basic Scatter Plot</a></li>
<li class="chapter" data-level="2.8.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#scatter-plot-with-trend-line"><i class="fa fa-check"></i><b>2.8.2</b> Scatter Plot with Trend Line</a></li>
<li class="chapter" data-level="2.8.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#boxplots"><i class="fa fa-check"></i><b>2.8.3</b> Boxplots</a></li>
<li class="chapter" data-level="2.8.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#histograms"><i class="fa fa-check"></i><b>2.8.4</b> Histograms</a></li>
<li class="chapter" data-level="2.8.5" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#density-plots"><i class="fa fa-check"></i><b>2.8.5</b> Density Plots</a></li>
<li class="chapter" data-level="2.8.6" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#venn-diagrams"><i class="fa fa-check"></i><b>2.8.6</b> Venn Diagrams</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exploratory-data-analysis-eda"><i class="fa fa-check"></i><b>2.9</b> Exploratory Data Analysis (EDA)</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-the-structure-of-the-data"><i class="fa fa-check"></i><b>2.9.1</b> Understanding the Structure of the Data</a></li>
<li class="chapter" data-level="2.9.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#checking-for-missing-values"><i class="fa fa-check"></i><b>2.9.2</b> Checking for Missing Values</a></li>
<li class="chapter" data-level="2.9.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#understanding-distributions"><i class="fa fa-check"></i><b>2.9.3</b> Understanding Distributions</a></li>
<li class="chapter" data-level="2.9.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#identifying-outliers"><i class="fa fa-check"></i><b>2.9.4</b> Identifying Outliers</a></li>
<li class="chapter" data-level="2.9.5" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#comparing-groups"><i class="fa fa-check"></i><b>2.9.5</b> Comparing Groups</a></li>
<li class="chapter" data-level="2.9.6" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exploring-relationships-between-variables"><i class="fa fa-check"></i><b>2.9.6</b> Exploring Relationships Between Variables</a></li>
<li class="chapter" data-level="2.9.7" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#simple-group-summaries"><i class="fa fa-check"></i><b>2.9.7</b> Simple Group Summaries</a></li>
<li class="chapter" data-level="2.9.8" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#eda-vs-statistical-analysis"><i class="fa fa-check"></i><b>2.9.8</b> EDA vs Statistical Analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#hands-on-exercises-1"><i class="fa fa-check"></i><b>2.10</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="2.10.1" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-1-1"><i class="fa fa-check"></i><b>2.10.1</b> Exercise 1</a></li>
<li class="chapter" data-level="2.10.2" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-2-1"><i class="fa fa-check"></i><b>2.10.2</b> Exercise 2</a></li>
<li class="chapter" data-level="2.10.3" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-3-1"><i class="fa fa-check"></i><b>2.10.3</b> Exercise 3</a></li>
<li class="chapter" data-level="2.10.4" data-path="day-2-data-manipulation-and-visualization-in-r.html"><a href="day-2-data-manipulation-and-visualization-in-r.html#exercise-4-challenge"><i class="fa fa-check"></i><b>2.10.4</b> Exercise 4 (Challenge)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html"><i class="fa fa-check"></i><b>3</b> Day-3 : Statistical Analysis Using R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#learning-objectives-3"><i class="fa fa-check"></i><b>3.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="3.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#types-of-data"><i class="fa fa-check"></i><b>3.2</b> Types of Data</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#why-do-we-need-statistics"><i class="fa fa-check"></i><b>3.2.1</b> Why Do We Need Statistics?</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#descriptive-vs-inferential-statistics"><i class="fa fa-check"></i><b>3.3</b> Descriptive vs Inferential Statistics</a></li>
<li class="chapter" data-level="3.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#descriptive-statistics"><i class="fa fa-check"></i><b>3.4</b> Descriptive Statistics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#measures-of-central-tendency"><i class="fa fa-check"></i><b>3.4.1</b> Measures of Central Tendency</a></li>
<li class="chapter" data-level="3.4.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#numeric-summaries"><i class="fa fa-check"></i><b>3.4.2</b> Numeric Summaries</a></li>
<li class="chapter" data-level="3.4.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#grouped-summaries"><i class="fa fa-check"></i><b>3.4.3</b> Grouped summaries:</a></li>
<li class="chapter" data-level="3.4.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#visualizing-distributions"><i class="fa fa-check"></i><b>3.4.4</b> Visualizing Distributions</a></li>
<li class="chapter" data-level="3.4.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#from-description-to-inference"><i class="fa fa-check"></i><b>3.4.5</b> From Description to Inference</a></li>
<li class="chapter" data-level="3.4.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#populations-vs-samples"><i class="fa fa-check"></i><b>3.4.6</b> Populations vs Samples</a></li>
<li class="chapter" data-level="3.4.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#sampling-distribution"><i class="fa fa-check"></i><b>3.4.7</b> Sampling Distribution</a></li>
<li class="chapter" data-level="3.4.8" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#from-variation-to-statistical-testing"><i class="fa fa-check"></i><b>3.4.8</b> From Variation to Statistical Testing</a></li>
<li class="chapter" data-level="3.4.9" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#interpreting-p-values-common-mistakes"><i class="fa fa-check"></i><b>3.4.9</b> Interpreting p-values (Common Mistakes)</a></li>
<li class="chapter" data-level="3.4.10" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#confidence-interval-interpretation"><i class="fa fa-check"></i><b>3.4.10</b> Confidence Interval Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#introduction-to-hypothesis-testing"><i class="fa fa-check"></i><b>3.5</b> Introduction to Hypothesis Testing</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#key-concepts"><i class="fa fa-check"></i><b>3.5.1</b> Key concepts:</a></li>
<li class="chapter" data-level="3.5.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#t-test"><i class="fa fa-check"></i><b>3.5.2</b> t-test</a></li>
<li class="chapter" data-level="3.5.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>3.5.3</b> Two-Sample t-test</a></li>
<li class="chapter" data-level="3.5.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#non-parametric-tests-brief-overview"><i class="fa fa-check"></i><b>3.5.4</b> Non-Parametric Tests (Brief Overview)</a></li>
<li class="chapter" data-level="3.5.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#analysis-of-variance-anova"><i class="fa fa-check"></i><b>3.5.5</b> Analysis of Variance (ANOVA)</a></li>
<li class="chapter" data-level="3.5.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#correlation-analysis"><i class="fa fa-check"></i><b>3.5.6</b> Correlation Analysis</a></li>
<li class="chapter" data-level="3.5.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#linear-regression"><i class="fa fa-check"></i><b>3.5.7</b> Linear Regression</a></li>
<li class="chapter" data-level="3.5.8" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#visualizing-the-regression"><i class="fa fa-check"></i><b>3.5.8</b> Visualizing the Regression</a></li>
<li class="chapter" data-level="3.5.9" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#interpreting-statistical-results"><i class="fa fa-check"></i><b>3.5.9</b> Interpreting Statistical Results</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#t-statistic-vs-effect-size-cohens-d-the-math"><i class="fa fa-check"></i><b>3.6</b> t-Statistic vs Effect Size (Cohen’s d): The Math</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#two-sample-t-statistic-welch-t-test"><i class="fa fa-check"></i><b>3.6.1</b> 1️⃣ Two-Sample t-Statistic (Welch t-test)</a></li>
<li class="chapter" data-level="3.6.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#effect-size-cohens-d"><i class="fa fa-check"></i><b>3.6.2</b> 2️⃣ Effect Size: Cohen’s d</a></li>
<li class="chapter" data-level="3.6.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#important-takeaway"><i class="fa fa-check"></i><b>3.6.3</b> Important Takeaway</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#machine-learning-statistical-learning-a-gentle-introduction"><i class="fa fa-check"></i><b>3.7</b> Machine Learning (Statistical Learning): A Gentle Introduction</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#statistics-vs-machine-learning-conceptual-view"><i class="fa fa-check"></i><b>3.7.1</b> Statistics vs Machine Learning (Conceptual View)</a></li>
<li class="chapter" data-level="3.7.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#a-simple-example-linear-regression-as-machine-learning"><i class="fa fa-check"></i><b>3.7.2</b> A Simple Example: Linear Regression as Machine Learning</a></li>
<li class="chapter" data-level="3.7.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#evaluating-predictions"><i class="fa fa-check"></i><b>3.7.3</b> Evaluating Predictions</a></li>
<li class="chapter" data-level="3.7.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#types-of-machine-learning-tasks"><i class="fa fa-check"></i><b>3.7.4</b> Types of Machine Learning Tasks</a></li>
<li class="chapter" data-level="3.7.5" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#regression-tasks"><i class="fa fa-check"></i><b>3.7.5</b> Regression Tasks</a></li>
<li class="chapter" data-level="3.7.6" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#classification-tasks"><i class="fa fa-check"></i><b>3.7.6</b> Classification Tasks</a></li>
<li class="chapter" data-level="3.7.7" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#advanced-machine-learning-overview"><i class="fa fa-check"></i><b>3.7.7</b> Advanced Machine Learning (Overview)</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#hands-on-exercises-2"><i class="fa fa-check"></i><b>3.8</b> Hands-on Exercises</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-1-2"><i class="fa fa-check"></i><b>3.8.1</b> Exercise 1</a></li>
<li class="chapter" data-level="3.8.2" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-2-2"><i class="fa fa-check"></i><b>3.8.2</b> Exercise 2</a></li>
<li class="chapter" data-level="3.8.3" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-3-2"><i class="fa fa-check"></i><b>3.8.3</b> Exercise 3</a></li>
<li class="chapter" data-level="3.8.4" data-path="day-3-statistical-analysis-using-r.html"><a href="day-3-statistical-analysis-using-r.html#exercise-4-challenge-1"><i class="fa fa-check"></i><b>3.8.4</b> Exercise 4 (Challenge)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html"><i class="fa fa-check"></i><b>4</b> Day-4 : Genomic Data Exploration Using R - 1</a>
<ul>
<li class="chapter" data-level="4.1" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#learning-objectives-4"><i class="fa fa-check"></i><b>4.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="4.2" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#what-is-genomic-data"><i class="fa fa-check"></i><b>4.2</b> What Is Genomic Data?</a></li>
<li class="chapter" data-level="4.3" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#what-is-bioconductor"><i class="fa fa-check"></i><b>4.3</b> What Is Bioconductor?</a></li>
<li class="chapter" data-level="4.4" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#installing-bioconductor"><i class="fa fa-check"></i><b>4.4</b> Installing Bioconductor</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#log-transformation-for-exploration"><i class="fa fa-check"></i><b>4.4.1</b> Log Transformation for Exploration</a></li>
<li class="chapter" data-level="4.4.2" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#differential-expression-analysis-conceptual"><i class="fa fa-check"></i><b>4.4.2</b> Differential Expression Analysis (Conceptual)</a></li>
<li class="chapter" data-level="4.4.3" data-path="day-4-genomic-data-exploration-using-r---1.html"><a href="day-4-genomic-data-exploration-using-r---1.html#volcano-plot"><i class="fa fa-check"></i><b>4.4.3</b> Volcano Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="day-5-genomic-data-exploration-using-r---2.html"><a href="day-5-genomic-data-exploration-using-r---2.html"><i class="fa fa-check"></i><b>5</b> Day-5 : Genomic Data Exploration Using R - 2</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Programming for Statistical &amp; Genomic Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="day-4-genomic-data-exploration-using-r---1" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Day-4 : Genomic Data Exploration Using R - 1<a href="day-4-genomic-data-exploration-using-r---1.html#day-4-genomic-data-exploration-using-r---1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="learning-objectives-4" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Learning Objectives<a href="day-4-genomic-data-exploration-using-r---1.html#learning-objectives-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>By the end of this day, students will be able to:</p>
<ul>
<li>Understand common genomic data types</li>
<li>Learn what Bioconductor is and why it is used</li>
<li>Work with expression data in R</li>
<li>Use Bioconductor data structures</li>
<li>Perform basic exploratory analysis on genomic datasets</li>
</ul>
<hr />
</div>
<div id="what-is-genomic-data" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> What Is Genomic Data?<a href="day-4-genomic-data-exploration-using-r---1.html#what-is-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Genomic data refers to information derived from DNA, RNA, or related molecules.</p>
<p>Common genomic data types include:</p>
<ul>
<li><strong>Gene expression data</strong> (RNA-seq, microarrays)</li>
<li><strong>Genomic coordinates</strong> (BED, GFF, GRanges)</li>
<li><strong>Variant data</strong> (VCF)</li>
<li><strong>Annotation data</strong> (gene names, pathways)</li>
</ul>
<p>In this course, we focus on <strong>gene expression data</strong>.</p>
<hr />
</div>
<div id="what-is-bioconductor" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> What Is Bioconductor?<a href="day-4-genomic-data-exploration-using-r---1.html#what-is-bioconductor" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Bioconductor</strong> is a collection of R packages designed for:</p>
<ul>
<li>Genomic data analysis</li>
<li>Biological annotation</li>
<li>Reproducible research</li>
</ul>
<p>Key features:</p>
<ul>
<li>Standardized data structures</li>
<li>Well-documented packages</li>
<li>Strong integration with R statistics</li>
</ul>
<p>Bioconductor complements CRAN.</p>
<hr />
</div>
<div id="installing-bioconductor" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Installing Bioconductor<a href="day-4-genomic-data-exploration-using-r---1.html#installing-bioconductor" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Install Bioconductor (only once):</p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb562-1" tabindex="-1"></a><span class="co"># install.packages(&quot;BiocManager&quot;) [unhash it for first run]</span></span>
<span id="cb562-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb562-2" tabindex="-1"></a></span>
<span id="cb562-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb562-3" tabindex="-1"></a><span class="co"># BiocManager::install() [unhash it for first run]</span></span></code></pre></div>
<p>Install required packages:</p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb563-1" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;SummarizedExperiment&quot;, &quot;airway&quot;, &quot;Biobase&quot;)) [unhash it for first run]</span></span></code></pre></div>
<p>Loading Required Libraries</p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb564-1" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     count</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: generics</code></pre>
<pre><code>## 
## Attaching package: &#39;generics&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:caret&#39;:
## 
##     train</code></pre>
<pre><code>## The following object is masked from &#39;package:lubridate&#39;:
## 
##     as.difftime</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     explain</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
##     setequal, union</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
##     unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     second, second&lt;-</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     findMatches</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:lubridate&#39;:
## 
##     %within%</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce</code></pre>
<pre><code>## Loading required package: Seqinfo</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb602-1" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb602-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb602-2" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb602-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb602-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb602-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb602-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb602-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb602-5" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<p>Example Dataset: airway</p>
<p>The airway dataset contains RNA-seq gene expression data from an experiment
studying the effect of dexamethasone on human airway smooth muscle cells.</p>
<p>What Kind of Data Is This?</p>
<ul>
<li><p>The airway dataset contains raw RNA-seq count data.</p></li>
<li><p>Each value represents the number of sequencing reads mapped to a gene</p></li>
<li><p>Values are integers</p></li>
<li><p>Data is not normalized</p></li>
<li><p>Data is not log-transformed</p></li>
</ul>
<p>Load the dataset:</p>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb603-1" tabindex="-1"></a><span class="fu">data</span>(airway)</span>
<span id="cb603-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb603-2" tabindex="-1"></a>airway</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 63677 8 
## metadata(1): &#39;&#39;
## assays(1): counts
## rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492
##   ENSG00000273493
## rowData names(10): gene_id gene_name ... seq_coord_system symbol
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<p>Understanding SummarizedExperiment</p>
<p>SummarizedExperiment is a core Bioconductor data structure.</p>
<p>It contains:</p>
<ul>
<li><p>assay data – expression matrix</p></li>
<li><p>rowData – gene-level information</p></li>
<li><p>colData – sample-level metadata</p></li>
</ul>
<p>Exploring the Expression Data
Assay Data (Counts Matrix)</p>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb605-1" tabindex="-1"></a><span class="fu">assay</span>(airway)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78</code></pre>
<p>Dimensions:</p>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb607-1" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">assay</span>(airway))</span></code></pre></div>
<pre><code>## [1] 63677     8</code></pre>
<p>Sample Metadata (colData)</p>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb609-1" tabindex="-1"></a><span class="fu">colData</span>(airway)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<p>Convert to data frame:</p>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb611-1" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(airway))</span>
<span id="cb611-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb611-2" tabindex="-1"></a><span class="fu">head</span>(sample_info)</span></code></pre></div>
<pre><code>##            SampleName    cell   dex albut        Run avgLength Experiment
## SRR1039508 GSM1275862  N61311 untrt untrt SRR1039508       126  SRX384345
## SRR1039509 GSM1275863  N61311   trt untrt SRR1039509       126  SRX384346
## SRR1039512 GSM1275866 N052611 untrt untrt SRR1039512       126  SRX384349
## SRR1039513 GSM1275867 N052611   trt untrt SRR1039513        87  SRX384350
## SRR1039516 GSM1275870 N080611 untrt untrt SRR1039516       120  SRX384353
## SRR1039517 GSM1275871 N080611   trt untrt SRR1039517       126  SRX384354
##               Sample    BioSample
## SRR1039508 SRS508568 SAMN02422669
## SRR1039509 SRS508567 SAMN02422675
## SRR1039512 SRS508571 SAMN02422678
## SRR1039513 SRS508572 SAMN02422670
## SRR1039516 SRS508575 SAMN02422682
## SRR1039517 SRS508576 SAMN02422673</code></pre>
<p>Gene Metadata (rowData)</p>
<div class="sourceCode" id="cb613"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb613-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb613-1" tabindex="-1"></a><span class="fu">rowData</span>(airway)</span></code></pre></div>
<pre><code>## DataFrame with 63677 rows and 10 columns
##                         gene_id     gene_name  entrezid   gene_biotype
##                     &lt;character&gt;   &lt;character&gt; &lt;integer&gt;    &lt;character&gt;
## ENSG00000000003 ENSG00000000003        TSPAN6        NA protein_coding
## ENSG00000000005 ENSG00000000005          TNMD        NA protein_coding
## ENSG00000000419 ENSG00000000419          DPM1        NA protein_coding
## ENSG00000000457 ENSG00000000457         SCYL3        NA protein_coding
## ENSG00000000460 ENSG00000000460      C1orf112        NA protein_coding
## ...                         ...           ...       ...            ...
## ENSG00000273489 ENSG00000273489 RP11-180C16.1        NA      antisense
## ENSG00000273490 ENSG00000273490        TSEN34        NA protein_coding
## ENSG00000273491 ENSG00000273491  RP11-138A9.2        NA        lincRNA
## ENSG00000273492 ENSG00000273492    AP000230.1        NA        lincRNA
## ENSG00000273493 ENSG00000273493  RP11-80H18.4        NA        lincRNA
##                 gene_seq_start gene_seq_end              seq_name seq_strand
##                      &lt;integer&gt;    &lt;integer&gt;           &lt;character&gt;  &lt;integer&gt;
## ENSG00000000003       99883667     99894988                     X         -1
## ENSG00000000005       99839799     99854882                     X          1
## ENSG00000000419       49551404     49575092                    20         -1
## ENSG00000000457      169818772    169863408                     1         -1
## ENSG00000000460      169631245    169823221                     1          1
## ...                        ...          ...                   ...        ...
## ENSG00000273489      131178723    131182453                     7         -1
## ENSG00000273490       54693789     54697585 HSCHR19LRC_LRC_J_CTG1          1
## ENSG00000273491      130600118    130603315          HG1308_PATCH          1
## ENSG00000273492       27543189     27589700                    21          1
## ENSG00000273493       58315692     58315845                     3          1
##                 seq_coord_system        symbol
##                        &lt;integer&gt;   &lt;character&gt;
## ENSG00000000003               NA        TSPAN6
## ENSG00000000005               NA          TNMD
## ENSG00000000419               NA          DPM1
## ENSG00000000457               NA         SCYL3
## ENSG00000000460               NA      C1orf112
## ...                          ...           ...
## ENSG00000273489               NA RP11-180C16.1
## ENSG00000273490               NA        TSEN34
## ENSG00000273491               NA  RP11-138A9.2
## ENSG00000273492               NA    AP000230.1
## ENSG00000273493               NA  RP11-80H18.4</code></pre>
<p>Basic Quality Checks
Library Sizes</p>
<div class="sourceCode" id="cb615"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb615-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb615-1" tabindex="-1"></a>library_sizes <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="fu">assay</span>(airway))</span>
<span id="cb615-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb615-2" tabindex="-1"></a>library_sizes</span></code></pre></div>
<pre><code>## SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 
##   20637971   18809481   25348649   15163415   24448408   30818215   19126151 
## SRR1039521 
##   21164133</code></pre>
<p>Visualize library sizes:</p>
<div class="sourceCode" id="cb617"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb617-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">data.frame</span>(library_sizes), <span class="fu">aes</span>(<span class="at">x =</span> library_sizes)) <span class="sc">+</span></span>
<span id="cb617-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb617-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-3" tabindex="-1"></a><span class="fu">labs</span>(</span>
<span id="cb617-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-4" tabindex="-1"></a><span class="at">title =</span> <span class="st">&quot;Library Size Distribution&quot;</span>,</span>
<span id="cb617-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-5" tabindex="-1"></a><span class="at">x =</span> <span class="st">&quot;Total Read Counts&quot;</span>,</span>
<span id="cb617-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-6" tabindex="-1"></a><span class="at">y =</span> <span class="st">&quot;Number of Samples&quot;</span></span>
<span id="cb617-7"><a href="day-4-genomic-data-exploration-using-r---1.html#cb617-7" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-135-1.png" alt="" width="672" /></p>
<div class="sourceCode" id="cb618"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb618-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-1" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(airway)</span>
<span id="cb618-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-2" tabindex="-1"></a></span>
<span id="cb618-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-3" tabindex="-1"></a><span class="fu">hist</span>(counts[, <span class="dv">1</span>],</span>
<span id="cb618-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-4" tabindex="-1"></a><span class="at">breaks =</span> <span class="dv">50</span>,</span>
<span id="cb618-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-5" tabindex="-1"></a><span class="at">main =</span> <span class="st">&quot;Distribution of Raw Counts (Sample 1)&quot;</span>,</span>
<span id="cb618-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb618-6" tabindex="-1"></a><span class="at">xlab =</span> <span class="st">&quot;Read Counts&quot;</span>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-136-1.png" alt="" width="672" /></p>
<div id="log-transformation-for-exploration" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Log Transformation for Exploration<a href="day-4-genomic-data-exploration-using-r---1.html#log-transformation-for-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Raw counts are highly skewed.</li>
<li>For visualization only, we use a log transformation:</li>
</ul>
<div class="sourceCode" id="cb619"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb619-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-1" tabindex="-1"></a>log_counts <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb619-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-2" tabindex="-1"></a></span>
<span id="cb619-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-3" tabindex="-1"></a><span class="fu">hist</span>(log_counts[, <span class="dv">1</span>],</span>
<span id="cb619-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-4" tabindex="-1"></a><span class="at">breaks =</span> <span class="dv">50</span>,</span>
<span id="cb619-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-5" tabindex="-1"></a><span class="at">main =</span> <span class="st">&quot;Log-transformed Counts (Sample 1)&quot;</span>,</span>
<span id="cb619-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb619-6" tabindex="-1"></a><span class="at">xlab =</span> <span class="st">&quot;log2(count + 1)&quot;</span>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-137-1.png" alt="" width="672" /></p>
<div class="sourceCode" id="cb620"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb620-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb620-1" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">colData</span>(airway)<span class="sc">$</span>dex</span>
<span id="cb620-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb620-2" tabindex="-1"></a>group</span></code></pre></div>
<pre><code>## [1] untrt trt   untrt trt   untrt trt   untrt trt  
## Levels: trt untrt</code></pre>
<div class="sourceCode" id="cb622"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb622-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-1" tabindex="-1"></a><span class="fu">boxplot</span>(</span>
<span id="cb622-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-2" tabindex="-1"></a>log_counts,</span>
<span id="cb622-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-3" tabindex="-1"></a><span class="at">col =</span> <span class="fu">ifelse</span>(group <span class="sc">==</span> <span class="st">&quot;trt&quot;</span>, <span class="st">&quot;tomato&quot;</span>, <span class="st">&quot;lightblue&quot;</span>),</span>
<span id="cb622-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-4" tabindex="-1"></a><span class="at">las =</span> <span class="dv">2</span>,</span>
<span id="cb622-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-5" tabindex="-1"></a><span class="at">main =</span> <span class="st">&quot;Log-transformed Expression Values&quot;</span></span>
<span id="cb622-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb622-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-139-1.png" alt="" width="672" /></p>
</div>
<div id="differential-expression-analysis-conceptual" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Differential Expression Analysis (Conceptual)<a href="day-4-genomic-data-exploration-using-r---1.html#differential-expression-analysis-conceptual" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>We now demonstrate how genomic analyses are typically performed using a specialized package.</li>
</ul>
<div class="sourceCode" id="cb623"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb623-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb623-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(</span>
<span id="cb623-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb623-2" tabindex="-1"></a>airway,</span>
<span id="cb623-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb623-3" tabindex="-1"></a><span class="at">design =</span> <span class="sc">~</span> dex</span>
<span id="cb623-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb623-4" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb624"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb624-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb624-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb631"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb631-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb631-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb631-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb631-2" tabindex="-1"></a></span>
<span id="cb631-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb631-3" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): dex untrt vs trt 
## Wald test p-value: dex untrt vs trt 
## DataFrame with 6 rows and 6 columns
##                   baseMean log2FoldChange     lfcSE      stat    pvalue
##                  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## ENSG00000000003 708.602170      0.3788470  0.173141  2.188082 0.0286636
## ENSG00000000005   0.000000             NA        NA        NA        NA
## ENSG00000000419 520.297901     -0.2037604  0.100599 -2.025478 0.0428183
## ENSG00000000457 237.163037     -0.0340428  0.126279 -0.269584 0.7874802
## ENSG00000000460  57.932633      0.1171786  0.301237  0.388992 0.6972820
## ENSG00000000938   0.318098      1.7245505  3.493633  0.493627 0.6215698
##                      padj
##                 &lt;numeric&gt;
## ENSG00000000003  0.139308
## ENSG00000000005        NA
## ENSG00000000419  0.183359
## ENSG00000000457  0.930572
## ENSG00000000460  0.895441
## ENSG00000000938        NA</code></pre>
<div class="sourceCode" id="cb633"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb633-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-1" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb633-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-2" tabindex="-1"></a>res_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df)</span>
<span id="cb633-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-3" tabindex="-1"></a></span>
<span id="cb633-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-4" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> res_df <span class="sc">%&gt;%</span></span>
<span id="cb633-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-5" tabindex="-1"></a><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb633-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-6" tabindex="-1"></a></span>
<span id="cb633-7"><a href="day-4-genomic-data-exploration-using-r---1.html#cb633-7" tabindex="-1"></a><span class="fu">nrow</span>(sig_genes)</span></code></pre></div>
<pre><code>## [1] 877</code></pre>
<div class="sourceCode" id="cb635"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb635-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb635-1" tabindex="-1"></a><span class="fu">head</span>(sig_genes)</span></code></pre></div>
<pre><code>##                   baseMean log2FoldChange     lfcSE      stat       pvalue
## ENSG00000003402 2546.61420      -1.183425 0.1635592 -7.235455 4.639712e-13
## ENSG00000004799  914.37901      -2.544032 0.9011874 -2.822977 4.757993e-03
## ENSG00000004846   17.99917       1.881406 0.6992853  2.690470 7.135153e-03
## ENSG00000005471   33.66399       1.218750 0.4343571  2.805871 5.018075e-03
## ENSG00000006788   10.25338      -3.169070 1.0663557 -2.971869 2.959927e-03
## ENSG00000008256 3716.57738      -1.204606 0.2043656 -5.894367 3.761198e-09
##                         padj            gene
## ENSG00000003402 5.563164e-11 ENSG00000003402
## ENSG00000004799 3.621101e-02 ENSG00000004799
## ENSG00000004846 4.938802e-02 ENSG00000004846
## ENSG00000005471 3.772691e-02 ENSG00000005471
## ENSG00000006788 2.511883e-02 ENSG00000006788
## ENSG00000008256 1.925671e-07 ENSG00000008256</code></pre>
</div>
<div id="volcano-plot" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> Volcano Plot<a href="day-4-genomic-data-exploration-using-r---1.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>A volcano plot shows both effect size and significance.</li>
</ul>
<div class="sourceCode" id="cb637"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb637-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-1" tabindex="-1"></a><span class="fu">ggplot</span>(res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb637-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-2" tabindex="-1"></a><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb637-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-3" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb637-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-4" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb637-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-5" tabindex="-1"></a><span class="fu">labs</span>(</span>
<span id="cb637-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-6" tabindex="-1"></a><span class="at">title =</span> <span class="st">&quot;Volcano Plot&quot;</span>,</span>
<span id="cb637-7"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-7" tabindex="-1"></a><span class="at">x =</span> <span class="st">&quot;Log2 Fold Change&quot;</span>,</span>
<span id="cb637-8"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-8" tabindex="-1"></a><span class="at">y =</span> <span class="st">&quot;-log10(p-value)&quot;</span></span>
<span id="cb637-9"><a href="day-4-genomic-data-exploration-using-r---1.html#cb637-9" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Warning: Removed 30259 rows containing missing values or values outside
## the scale range (`geom_point()`).</code></pre>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-143-1.png" alt="" width="672" /></p>
<ul>
<li>we can add colors based on up-regulation and down-regulation.</li>
</ul>
<div class="sourceCode" id="cb639"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb639-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-1" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> res_df <span class="sc">%&gt;%</span></span>
<span id="cb639-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb639-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-3" tabindex="-1"></a>    <span class="at">regulation =</span> <span class="fu">case_when</span>(</span>
<span id="cb639-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-4" tabindex="-1"></a>      padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>  <span class="sc">~</span> <span class="st">&quot;Up-regulated&quot;</span>,</span>
<span id="cb639-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-5" tabindex="-1"></a>      padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;Down-regulated&quot;</span>,</span>
<span id="cb639-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-6" tabindex="-1"></a>      <span class="cn">TRUE</span>                             <span class="sc">~</span> <span class="st">&quot;Not significant&quot;</span></span>
<span id="cb639-7"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-7" tabindex="-1"></a>    )</span>
<span id="cb639-8"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-8" tabindex="-1"></a>  )</span>
<span id="cb639-9"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-9" tabindex="-1"></a></span>
<span id="cb639-10"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-10" tabindex="-1"></a><span class="fu">ggplot</span>(res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), <span class="at">color =</span> regulation)) <span class="sc">+</span></span>
<span id="cb639-11"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb639-12"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-12" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb639-13"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-13" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb639-14"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-14" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb639-15"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-15" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb639-16"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-16" tabindex="-1"></a>      <span class="st">&quot;Up-regulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb639-17"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-17" tabindex="-1"></a>      <span class="st">&quot;Down-regulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb639-18"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-18" tabindex="-1"></a>      <span class="st">&quot;Not significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span></span>
<span id="cb639-19"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-19" tabindex="-1"></a>    )</span>
<span id="cb639-20"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-20" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb639-21"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-21" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb639-22"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-22" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Volcano Plot of Differential Gene Expression&quot;</span>,</span>
<span id="cb639-23"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-23" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Log2 Fold Change&quot;</span>,</span>
<span id="cb639-24"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-24" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;-log10(p-value)&quot;</span>,</span>
<span id="cb639-25"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-25" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Gene Regulation&quot;</span></span>
<span id="cb639-26"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-26" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb639-27"><a href="day-4-genomic-data-exploration-using-r---1.html#cb639-27" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<pre><code>## Warning: Removed 30259 rows containing missing values or values outside
## the scale range (`geom_point()`).</code></pre>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-144-1.png" alt="" width="672" /></p>
<ul>
<li>adding the labels for top most significant genes.</li>
</ul>
<div class="sourceCode" id="cb641"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb641-1"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-1" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb641-2"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-2" tabindex="-1"></a></span>
<span id="cb641-3"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-3" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> res_df <span class="sc">%&gt;%</span></span>
<span id="cb641-4"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-4" tabindex="-1"></a>  <span class="fu">filter</span>(regulation <span class="sc">!=</span> <span class="st">&quot;Not significant&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb641-5"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-5" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb641-6"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-6" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb641-7"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-7" tabindex="-1"></a></span>
<span id="cb641-8"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-8" tabindex="-1"></a><span class="fu">ggplot</span>(res_df, <span class="fu">aes</span>(log2FoldChange, <span class="sc">-</span><span class="fu">log10</span>(pvalue), <span class="at">color =</span> regulation)) <span class="sc">+</span></span>
<span id="cb641-9"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb641-10"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-10" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb641-11"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-11" tabindex="-1"></a>    <span class="at">data =</span> top_genes,</span>
<span id="cb641-12"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-12" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> gene),</span>
<span id="cb641-13"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-13" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span></span>
<span id="cb641-14"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-14" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb641-15"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-15" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb641-16"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-16" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb641-17"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-17" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb641-18"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-18" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Up-regulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb641-19"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-19" tabindex="-1"></a>               <span class="st">&quot;Down-regulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb641-20"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-20" tabindex="-1"></a>               <span class="st">&quot;Not significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span>)</span>
<span id="cb641-21"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-21" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb641-22"><a href="day-4-genomic-data-exploration-using-r---1.html#cb641-22" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<pre><code>## Warning: Removed 30259 rows containing missing values or values outside
## the scale range (`geom_point()`).</code></pre>
<p><img src="R_Programming_Course_files/figure-html/unnamed-chunk-145-1.png" alt="" width="672" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="day-3-statistical-analysis-using-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="day-5-genomic-data-exploration-using-r---2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["R_Programming_Course.pdf", "R_Programming_Course.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
